// @jd/ant-theme@0.0.1-alpha.1/dist/Button/demo/ant-disabled.js
import { Button, Flex } from "../..";
import React from 'react';
var App = function App() {
  return /*#__PURE__*/React.createElement(Flex, {
    gap: "small",
    align: "flex-start",
    vertical: true
  }, /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "primary"
  }, "Primary"), /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    disabled: true
  }, "Primary(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, null, "Default"), /*#__PURE__*/React.createElement(Button, {
    disabled: true
  }, "Default(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "dashed"
  }, "Dashed"), /*#__PURE__*/React.createElement(Button, {
    type: "dashed",
    disabled: true
  }, "Dashed(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "text"
  }, "Text"), /*#__PURE__*/React.createElement(Button, {
    type: "text",
    disabled: true
  }, "Text(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "link"
  }, "Link"), /*#__PURE__*/React.createElement(Button, {
    type: "link",
    disabled: true
  }, "Link(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    href: "https://ant.design/index-cn"
  }, "Href Primary"), /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    href: "https://ant.design/index-cn",
    disabled: true
  }, "Href Primary(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    danger: true
  }, "Danger Default"), /*#__PURE__*/React.createElement(Button, {
    danger: true,
    disabled: true
  }, "Danger Default(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    danger: true,
    type: "text"
  }, "Danger Text"), /*#__PURE__*/React.createElement(Button, {
    danger: true,
    type: "text",
    disabled: true
  }, "Danger Text(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small"
  }, /*#__PURE__*/React.createElement(Button, {
    type: "link",
    danger: true
  }, "Danger Link"), /*#__PURE__*/React.createElement(Button, {
    type: "link",
    danger: true,
    disabled: true
  }, "Danger Link(disabled)")), /*#__PURE__*/React.createElement(Flex, {
    gap: "small",
    className: "bg-bg p-10"
  }, /*#__PURE__*/React.createElement(Button, {
    ghost: true
  }, "Ghost"), /*#__PURE__*/React.createElement(Button, {
    ghost: true,
    disabled: true
  }, "Ghost(disabled)")));
};
export default App;
