// @jd/ant-theme@0.0.1-alpha.1/dist/Dropdown/demo/basic.js
import { ErrorWrapper } from "../../components";
import { Button, Dropdown } from "../..";
import React from 'react';
import styles from "../../Button/demo/index.module.less";
var items = [{
  key: '1',
  label: /*#__PURE__*/React.createElement("a", {
    target: "_blank",
    rel: "noopener noreferrer",
    href: "https://www.antgroup.com"
  }, "1st menu item")
}, {
  key: '2',
  label: /*#__PURE__*/React.createElement("a", {
    target: "_blank",
    rel: "noopener noreferrer",
    href: "https://www.aliyun.com"
  }, "2nd menu item")
}, {
  key: '3',
  label: /*#__PURE__*/React.createElement("a", {
    target: "_blank",
    rel: "noopener noreferrer",
    href: "https://www.luohanacademy.com"
  }, "3rd menu item")
}];
/**
 *  基础按钮
 */
export default function () {
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full text-left")
  }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", {
    className: "whitespace-nowrap"
  }, /*#__PURE__*/React.createElement("td", null, "button\u7EC4\u4EF6\u5524\u8D77\u4E0B\u62C9\u83DC\u5355"), /*#__PURE__*/React.createElement("td", null, "\u6587\u5B57\u94FE\u6298\u53E0\uFF0C\u5524\u8D77\u4E0B\u62C9\u83DC\u5355\uFF1A"), /*#__PURE__*/React.createElement("td", null, "\u56FE\u6807\u6309\u94AE\uFF0C\u5524\u8D77\u4E0B\u62C9\u83DC\u5355"), /*#__PURE__*/React.createElement("td", null, "\u9F20\u6807\u53F3\u51FB\u5524\u8D77\u4E0B\u62C9\u83DC\u5355")), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: items
    },
    placement: "bottomLeft"
  }, /*#__PURE__*/React.createElement(Button, null, "bottomLeft")))))));
}
