// @jd/ant-theme@0.0.1-alpha.1/dist/Tag/demo/ant-borderless.js
import { Divider, Space, Tag } from "../..";
import React from 'react';
var App = function App() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Space, {
    size: [0, 'small'],
    wrap: true
  }, /*#__PURE__*/React.createElement(Tag, {
    bordered: false
  }, "Tag 1"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false
  }, "Tag 2"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    closable: true
  }, "Tag 3"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    closable: true
  }, "Tag 4")), /*#__PURE__*/React.createElement(Divider, null), /*#__PURE__*/React.createElement(Space, {
    size: [0, 'small'],
    wrap: true
  }, /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "processing"
  }, "processing"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "success"
  }, "success"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "error"
  }, "error"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "warning"
  }, "warning"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "magenta"
  }, "magenta"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "red"
  }, "red"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "volcano"
  }, "volcano"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "orange"
  }, "orange"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "gold"
  }, "gold"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "lime"
  }, "lime"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "green"
  }, "green"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "cyan"
  }, "cyan"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "blue"
  }, "blue"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "geekblue"
  }, "geekblue"), /*#__PURE__*/React.createElement(Tag, {
    bordered: false,
    color: "purple"
  }, "purple")));
};
export default App;
