// @jd/ant-theme@0.0.1-alpha.1/dist/Space/demo/element.js
import { ErrorWrapper, SpaceTip } from "../../components";
import { Col, Row } from "../..";
import React from 'react';
/**
 * 基础结构
 */
export default function () {
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "flex gap-16 flex-col"
  }, /*#__PURE__*/React.createElement("div", {
    className: "border-1 p-24 border-border rounded-card"
  }, /*#__PURE__*/React.createElement(Row, {
    gutter: 12
  }, new Array(3).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      key: index,
      span: 8
    }, /*#__PURE__*/React.createElement("div", {
      className: "h-106 rounded-card bg-sys-surface-surface-100 relative "
    }, index < 2 ? /*#__PURE__*/React.createElement(SpaceTip, {
      left: 12,
      leftTip: index === 0 ? '卡片内模块之间间距' : null,
      className: 'absolute h-full left-full'
    }) : null));
  }))), /*#__PURE__*/React.createElement("div", {
    className: "border-1 p-24 flex flex-col gap-12 border-border rounded-card relative"
  }, /*#__PURE__*/React.createElement("div", {
    className: "h-106 rounded-card bg-sys-surface-surface-100"
  }), /*#__PURE__*/React.createElement(SpaceTip, {
    className: "h-0 w-full -mt-12",
    top: 12,
    topTip: "\u5361\u7247\u5185\u6A21\u5757\u4E4B\u95F4\u95F4\u8DDD"
  }), /*#__PURE__*/React.createElement("div", {
    className: "h-106 rounded-card bg-sys-surface-surface-100"
  }), /*#__PURE__*/React.createElement(SpaceTip, {
    className: "w-full top-0 absolute -mt-17",
    top: 16,
    topTip: "\u697C\u5C42\u5361\u7247\u95F4\u8DDD"
  }))));
}
