// @jd/ant-theme@0.0.1-alpha.1/dist/Grid/demo/base.js
import { ErrorWrapper } from "../../components";
import { Col, Row, Tag, theme } from "../..";
import React from 'react';
import styles from "../../Button/demo/index.module.less";
var useToken = theme.useToken;

/**
 * 字体
 */
export default function () {
  var _useToken = useToken(),
    token = _useToken.token;
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "min-w-[1600px]"
  }, /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full")
  }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "24\u6805\u683C"), /*#__PURE__*/React.createElement("td", {
    className: "pt-80"
  }, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, new Array(24).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      span: 1,
      key: index
    }, /*#__PURE__*/React.createElement("div", {
      className: "w-full h-full bg-sys-color-primary-200 relative"
    }, index === 0 ? /*#__PURE__*/React.createElement("div", {
      className: "absolute w-full left-0 h-24 border-l-1 border-r-1 border-primary flex items-center justify-center"
    }, /*#__PURE__*/React.createElement("div", {
      className: "h-1 bg-primary flex-1"
    }), /*#__PURE__*/React.createElement(Tag, {
      className: "absolute bottom-full mr-0 mb-5 rounded-full text-12",
      color: token['colorPrimary'],
      size: "middle",
      bordered: false
    }, "Column")) : null, index === 1 ? /*#__PURE__*/React.createElement("div", {
      className: "absolute w-16 left-full h-24 border-l-1 border-r-1 border-sys-state-info flex items-center justify-center"
    }, /*#__PURE__*/React.createElement("div", {
      className: "h-1 bg-sys-state-info flex-1"
    }), /*#__PURE__*/React.createElement(Tag, {
      className: "absolute bottom-full mr-0 mb-5 rounded-full text-12"
      // color={token['sys-state-info']}
      ,
      size: "middle",
      bordered: false
    }, "Gutter")) : null));
  })))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "12\u6805\u683C"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, new Array(12).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      span: 2,
      key: index
    }, /*#__PURE__*/React.createElement("div", {
      className: "w-full h-full bg-colors-yellow-200"
    }));
  })))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "6\u6805\u683C"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, new Array(6).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      span: 4,
      key: index
    }, /*#__PURE__*/React.createElement("div", {
      className: "w-full h-full bg-colors-yellow-200"
    }));
  })))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "3\u6805\u683C"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, new Array(3).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      span: 8,
      key: index
    }, /*#__PURE__*/React.createElement("div", {
      className: "w-full h-full bg-colors-yellow-200"
    }));
  })))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "2\u6805\u683C"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, new Array(2).fill(null).map(function (item, index) {
    return /*#__PURE__*/React.createElement(Col, {
      span: 12,
      key: index
    }, /*#__PURE__*/React.createElement("div", {
      className: "w-full h-full bg-colors-yellow-200"
    }));
  })))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }, "\u901A\u680F"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, /*#__PURE__*/React.createElement(Col, {
    span: 24
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  }))))))), /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full mt-30")
  }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null), /*#__PURE__*/React.createElement("td", {
    className: "text-left"
  }, "\u975E\u7B49\u5206\u5206\u680F\u793A\u610F")), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, /*#__PURE__*/React.createElement(Col, {
    span: 6
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 12
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 6
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  }))))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, /*#__PURE__*/React.createElement(Col, {
    span: 8
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 16
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  }))))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, /*#__PURE__*/React.createElement(Col, {
    span: 12
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 6
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 6
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  }))))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "w-80"
  }), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Row, {
    gutter: 16,
    className: "h-120"
  }, /*#__PURE__*/React.createElement(Col, {
    span: 4
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 12
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })), /*#__PURE__*/React.createElement(Col, {
    span: 8
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-full h-full bg-colors-yellow-200"
  })))))))));
}
