// @jd/ant-theme@0.0.1-alpha.1/dist/Theme/ea/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { ErrorWrapper } from "../../components";
import { ConfigProvider } from "../..";
import { getProps } from "../../utils/PropsType";
import merge from 'lodash/merge';
import React from 'react';
import tokens, { formTokens } from "./token";
// import defaultToken from "../../CustomTheme/defaultToken"
// 扩展ant token 类型

var prefixCls = 'theme-ea';

/**
 * 主题配置
 */
var ThemeEa = function ThemeEa(props) {
  var _props$theme;
  // 合并外部tokens，方便内部使用
  var currentToken = merge(tokens, props === null || props === void 0 || (_props$theme = props.theme) === null || _props$theme === void 0 ? void 0 : _props$theme.token);
  console.log(currentToken, 8888);
  var wrapperProps = getProps(props, {
    prefixCls: prefixCls,
    theme: {
      cssVar: {
        key: prefixCls
      },
      components: {
        Button: {
          contentFontSizeLG: 14,
          contentFontSizeSM: 12,
          contentLineHeightLG: 1.4,
          contentLineHeight: 1.4,
          contentLineHeightSM: 1.4,
          paddingInlineLG: 14,
          paddingInlineSM: 8,
          paddingInline: 12,
          borderColorDisabled: '#dcdee0',
          defaultBorderColor: '#dcdee0',
          defaultColor: currentToken['colorText'] || '#1b1b1b',
          defaultGhostBorderColor: '#dcdee0',
          defaultGhostColor: currentToken['colorText'] || '#1b1b1b'
        },
        Tag: {
          defaultBg: '#f9f9f9'
        },
        Menu: {
          itemBg: currentToken['bg-page'] || '#F4F5F6',
          subMenuItemBg: 'transparent',
          itemSelectedBg: 'white',
          itemHoverBg: currentToken['sys-surface-outline-100'] || '#F0F1F2',
          itemActiveBg: 'rgba(255,255,255,0.4)',
          itemBorderRadius: 6,
          subMenuItemBorderRadius: 6,
          iconMarginInlineEnd: 8,
          iconSize: 16,
          activeBarBorderWidth: 0
        },
        Tabs: {
          itemColor: currentToken['sys-text-secondary'] || '#6A6A6A'
        },
        Breadcrumb: {
          separatorMargin: 14,
          linkHoverColor: currentToken["sys-color-primary-600"],
          linkColor: currentToken['sys-text-secondary'] || '#6A6A6A',
          separatorColor: currentToken['sys-text-disable'] || '#CECECE'
        },
        Input: _objectSpread({}, formTokens),
        InputNumber: _objectSpread({}, formTokens),
        Radio: {
          radioSize: 16
          // wrapperMarginInlineEnd: 36,
        }
      },
      token: currentToken
    }
  }, [], true);
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement(ConfigProvider, wrapperProps, props.children));
};
ThemeEa.defaultProps = {};
export default ThemeEa;
