// @jd/ant-theme@0.0.1-alpha.1/dist/Theme/ea/token.js
import defaultToken from "../../CustomTheme/defaultToken";
var tokens = {
  borderRadius: 4,
  controlHeightSM: 28,
  white: '#FFFFFF',
  borderRadiusCard: 12
};
// 散点的变量设置
var scatteredCssVar = {
  colorPrimary: 'sysColorPrimary600',
  colorText: 'colorsNeutral1000',
  colorSecondary: 'colorsNeutral700',
  colorTertiary: 'colorsNeutral500',
  colorTextDisable: 'colorsNeutral300',
  colorTextDisabled: 'colorsNeutral300',
  colorTextAnti: 'white',
  colorPrimaryHover: 'sysColorPrimary500',
  colorPrimaryActive: 'sysColorPrimary700',
  colorLink: 'sysStateLink',
  colorLinkHover: 'colorsBlue500',
  colorLinkActive: 'colorsBlue700',
  colorInfo: 'sysStateInfo',
  colorWarning: 'sysStateWarning',
  colorError: 'sysStateError',
  // 分割线
  colorSplit: 'sysSurfaceOutline006',
  colorBorder: 'sysSurfaceOutline200',
  colorBorderSecondary: 'sysSurfaceOutline006',
  colorTextPlaceholder: 'colorsNeutral400',
  colorBgContainerDisabled: 'sysSurfaceSurface100'
};
Object.keys(scatteredCssVar).forEach(function (e) {
  // @ts-ignore
  tokens[e] = defaultToken[scatteredCssVar[e]] || scatteredCssVar[e];
});
// shadow
tokens['shadow-mini'] = "0px 1px 3px rgba(0, 0, 0, 0.06),\n0px 2px 6px rgba(0, 0, 0, 0.04)";
tokens['shadow-low'] = "0px 4px 6px rgba(0, 0, 0, 0.06),\n0px 2px 12px rgba(0, 0, 0, 0.04)";
tokens['shadow-middle'] = "0px 12px 24px -4px rgba(0, 0, 0, 0.06),\n0px 4px 28px rgba(0, 0, 0, 0.04)";
tokens['shadow-middle'] = "0px 12px 24px -4px rgba(0, 0, 0, 0.06),\n0px 4px 28px rgba(0, 0, 0, 0.04)";
tokens['shadow-high'] = "0px 20px 35px -6px rgba(0, 0, 0, 0.06),\n0px 8px 56px -2px rgba(0, 0, 0, 0.04)";

// 字体设置
tokens['fontFamily'] = "JDZhengHei, PingFang SC, Microsoft Yahei, Lantinghei SC,\nHiragino Sans GB, 'Microsoft Sans Serif', WenQuanYi Micro Hei, sans";

// 字体大小
tokens['fontSizeLG'] = 14;

// size 大小
tokens['controlHeightSM'] = 28;
tokens['controlHeight'] = 32;
tokens['controlHeightLG'] = 40;

// placeholder
tokens['lineHeightXL'] = 24;
console.log(9399393, tokens);
export default tokens;

// padding
tokens['paddingXS'] = 6;
// console.log(tokens, 88888888);
export var formTokens = {
  activeBorderColor: tokens['colors-neutral-1000'],
  hoverBorderColor: tokens['colors-neutral-1000'],
  addonBg: '#fff',
  inputFontSizeLG: 16,
  inputFontSizeXL: 16,
  // activeShadow: '0 0 0 2px rgba(30,42,59,.2)',
  activeShadow: 'none',
  paddingInlineLG: 13,
  paddingInlineSM: 11,
  paddingInlineXL: 15,
  // paddingBlockXL: 13,
  // paddingBlockSM: 6,
  // paddingBlock: 8,
  // paddingBlockLG: 7,
  paddingBlockXL: 0,
  paddingBlockSM: 0,
  paddingBlock: 0,
  paddingBlockLG: 0
};
