// @kuma-ui/core@1.5.7/dist/components/componentList.mjs
// src/components/componentList.ts
import {
  isBoxProps,
  boxHandler,
  boxDefaultProps,
  defaultBoxTag
} from "./Box/handler.mjs";
import {
  defaultButtonTag,
  isButtonProps,
  buttonHandler,
  buttonDefaultProps
} from "./Button/handler.mjs";
import {
  defaultFlexTag,
  isFlexProps,
  flexHandler,
  flexDefaultProps
} from "./Flex/handler.mjs";
import {
  defaultHeadingTag,
  isHeadingProps,
  headingHandler,
  headingDefaultProps
} from "./Heading/handler.mjs";
import {
  defaultSpacerTag,
  isSpacerProps,
  spacerHandler,
  spacerDefaultProps
} from "./Spacer/handler.mjs";
import {
  defaultTextTag,
  isTextProps,
  textHandler,
  textDefaultProps
} from "./Text/handler.mjs";
import {
  defaultSelectTag,
  isSelectProps,
  selectHandler,
  selectDefaultProps
} from "./Select/handler.mjs";
import {
  defaultInputTag,
  isInputProps,
  inputHandler,
  inputDefaultProps
} from "./Input/handler.mjs";
import {
  defaultHStackTag,
  isHStackProps,
  hstackHandler,
  hstackDefaultProps
} from "./HStack/handler.mjs";
import {
  defaultVStackTag,
  isVStackProps,
  vstackHandler,
  vstackDefaultProps
} from "./VStack/handler.mjs";
import {
  defaultImageTag,
  isImageProps,
  imageHandler,
  imageDefaultProps
} from "./Image/handler.mjs";
import {
  defaultLinkTag,
  isLinkProps,
  linkHandler,
  linkDefaultProps
} from "./Link/handler.mjs";
import {
  defaultGridTag,
  isGridProps,
  gridHandler,
  gridDefaultProps
} from "./Grid/handler.mjs";
import "react";
var componentList = Object.freeze({
  Box: "Box",
  Flex: "Flex",
  Spacer: "Spacer",
  Text: "Text",
  Button: "Button",
  Heading: "Heading",
  Select: "Select",
  Input: "Input",
  HStack: "HStack",
  VStack: "VStack",
  Image: "Image",
  Link: "Link",
  Grid: "Grid"
});
var defaultComponentTag = {
  Box: defaultBoxTag,
  Button: defaultButtonTag,
  Flex: defaultFlexTag,
  Heading: defaultHeadingTag,
  Spacer: defaultSpacerTag,
  Text: defaultTextTag,
  Select: defaultSelectTag,
  Input: defaultInputTag,
  HStack: defaultHStackTag,
  VStack: defaultVStackTag,
  Image: defaultImageTag,
  Link: defaultLinkTag,
  Grid: defaultGridTag
};
var isComponentProps = (componentName) => (propName) => {
  switch (componentName) {
    case "Box":
      return isBoxProps(propName);
    case "Button":
      return isButtonProps(propName);
    case "Flex":
      return isFlexProps(propName);
    case "Heading":
      return isHeadingProps(propName);
    case "Spacer":
      return isSpacerProps(propName);
    case "Text":
      return isTextProps(propName);
    case "Select":
      return isSelectProps(propName);
    case "Input":
      return isInputProps(propName);
    case "HStack":
      return isHStackProps(propName);
    case "VStack":
      return isVStackProps(propName);
    case "Image":
      return isImageProps(propName);
    case "Link":
      return isLinkProps(propName);
    case "Grid":
      return isGridProps(propName);
    default:
      return componentName;
  }
};
var componentDefaultProps = (componentName) => {
  switch (componentName) {
    case "Box":
      return boxDefaultProps;
    case "Button":
      return buttonDefaultProps;
    case "Flex":
      return flexDefaultProps;
    case "Heading":
      return headingDefaultProps;
    case "Spacer":
      return spacerDefaultProps;
    case "Text":
      return textDefaultProps;
    case "Select":
      return selectDefaultProps;
    case "Input":
      return inputDefaultProps;
    case "HStack":
      return hstackDefaultProps;
    case "VStack":
      return vstackDefaultProps;
    case "Image":
      return imageDefaultProps;
    case "Link":
      return linkDefaultProps;
    case "Grid":
      return gridDefaultProps;
    default:
      return componentName;
  }
};
var componentHandler = (componentName) => (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- FIXME
  (props) => {
    switch (componentName) {
      case "Box":
        return boxHandler(props);
      case "Button":
        return buttonHandler(props);
      case "Flex":
        return flexHandler(props);
      case "Heading":
        return headingHandler(props);
      case "Spacer":
        return spacerHandler(props);
      case "Text":
        return textHandler(props);
      case "Select":
        return selectHandler(props);
      case "Input":
        return inputHandler(props);
      case "HStack":
        return hstackHandler(props);
      case "VStack":
        return vstackHandler(props);
      case "Image":
        return imageHandler(props);
      case "Link":
        return linkHandler(props);
      case "Grid":
        return gridHandler(props);
      default:
        return componentName;
    }
  }
);
export {
  componentDefaultProps,
  componentHandler,
  componentList,
  defaultComponentTag,
  isComponentProps
};
