// @jd/ant-theme@0.0.1-alpha.1/dist/CustomTheme/defaultToken.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import _kebabCase from 'lodash/kebabCase';
/** 基础颜色 */
var baseColorsVar = {
  colorsNeutral100: '#f3f3f3',
  colorsNeutral200: '#e7e7e7',
  colorsNeutral300: '#cecece',
  colorsNeutral400: '#B6B6B6',
  colorsNeutral500: '#9d9d9d',
  colorsNeutral600: '#858585',
  colorsNeutral700: '#6a6a6a',
  colorsNeutral800: '#505050',
  colorsNeutral900: '#353535',
  colorsNeutral1000: '#1b1b1b',
  colorsGray50: '#f8f8f9',
  colorsGray100: '#f0f1f2',
  colorsGray200: '#e1e3e6',
  colorsGray300: '#c4c7cc',
  colorsGray400: '#a6aab3',
  colorsGray500: '#898e99',
  colorsGray600: '#6b7280',
  colorsGray700: '#565b66',
  colorsGray800: '#40444d',
  colorsGray900: '#2b2e33',
  colorsGray1000: '#15171a',
  colorsGreen100: '#eaf8ec',
  colorsGreen200: '#d5f2da',
  colorsGreen300: '#ace4b4',
  colorsGreen400: '#82d78f',
  colorsGreen500: '#59c969',
  colorsGreen600: '#2fbc44',
  colorsGreen700: '#269636',
  colorsGreen800: '#1c7129',
  colorsGreen900: '#134b1b',
  colorsGreen1000: '#09260e',
  colorsCyan100: '#e8fafd',
  colorsCyan200: '#d1f5fb',
  colorsCyan300: '#a3ebf8',
  colorsCyan400: '#76e1f4',
  colorsCyan500: '#48d7f1',
  colorsCyan600: '#1acded',
  colorsCyan700: '#15a4be',
  colorsCyan800: '#107b8e',
  colorsCyan900: '#0a525f',
  colorsCyan1000: '#05292f',
  colorsSkyBlue100: '#eaf5fd',
  colorsSkyBlue200: '#d4ecfb',
  colorsSkyBlue300: '#a9d9f7',
  colorsSkyBlue400: '#7ec5f2',
  colorsSkyBlue500: '#53b2ee',
  colorsSkyBlue600: '#289fea',
  colorsSkyBlue700: '#207fbb',
  colorsSkyBlue800: '#185f8c',
  colorsSkyBlue900: '#10405e',
  colorsSkyBlue1000: '#08202f',
  colorsBlue100: '#e8f0fe',
  colorsBlue200: '#d1e1fd',
  colorsBlue300: '#a3c3fb',
  colorsBlue400: '#74a5f9',
  colorsBlue500: '#4687f7',
  colorsBlue600: '#1869f5',
  colorsBlue700: '#1354c4',
  colorsBlue800: '#0e3f93',
  colorsBlue900: '#0a2a62',
  colorsBlue1000: '#051531',
  colorsIndigo100: '#edeffd',
  colorsIndigo200: '#dbdffa',
  colorsIndigo300: '#b7bff6',
  colorsIndigo400: '#929ff1',
  colorsIndigo500: '#6e7fed',
  colorsIndigo600: '#4a5fe8',
  colorsIndigo700: '#3b4cba',
  colorsIndigo800: '#2c398b',
  colorsIndigo900: '#1e265d',
  colorsIndigo1000: '#0f132e',
  colorsPurple100: '#f2eafe',
  colorsPurple200: '#e6d6fd',
  colorsPurple300: '#ccadfb',
  colorsPurple400: '#b383f9',
  colorsPurple500: '#995af7',
  colorsPurple600: '#8031f5',
  colorsPurple700: '#6627c4',
  colorsPurple800: '#4d1d93',
  colorsPurple900: '#331462',
  colorsPurple1000: '#1a0a31',
  colorsFuchsia100: '#f9ecfb',
  colorsFuchsia200: '#f3d8f8',
  colorsFuchsia300: '#e6b1f1',
  colorsFuchsia400: '#da8be9',
  colorsFuchsia500: '#cd64e2',
  colorsFuchsia600: '#c13ddb',
  colorsFuchsia700: '#9a31af',
  colorsFuchsia800: '#742583',
  colorsFuchsia900: '#4d1858',
  colorsFuchsia1000: '#270c2c',
  colorsPink100: '#feebf3',
  colorsPink200: '#fdd7e8',
  colorsPink300: '#fcafd0',
  colorsPink400: '#fa86b9',
  colorsPink500: '#f95ea1',
  colorsPink600: '#f7368a',
  colorsPink700: '#c62b6e',
  colorsPink800: '#942053',
  colorsPink900: '#631637',
  colorsPink1000: '#310b1c',
  colorsRed50: '#fff5f4',
  colorsRed100: '#feebe8',
  colorsRed200: '#fdd6d1',
  colorsRed300: '#fbada3',
  colorsRed400: '#fa8474',
  colorsRed500: '#f85b46',
  colorsRed600: '#f63218',
  colorsRed700: '#c52813',
  colorsRed800: '#941e0e',
  colorsRed900: '#62140a',
  colorsRed1000: '#310a05',
  colorsOrange100: '#feeee8',
  colorsOrange200: '#fcdcd1',
  colorsOrange300: '#fabaa3',
  colorsOrange400: '#f79774',
  colorsOrange500: '#f95ea1',
  colorsOrange600: '#f25218',
  colorsOrange700: '#c24213',
  colorsOrange800: '#91310e',
  colorsOrange900: '#61210a',
  colorsOrange1000: '#301005',
  colorsYellow100: '#fff8ea',
  colorsYellow200: '#fef1d6',
  colorsYellow300: '#fde2ad',
  colorsYellow400: '#fdd483',
  colorsYellow500: '#fcc55a',
  colorsYellow600: '#fbb731',
  colorsYellow700: '#c99227',
  colorsYellow800: '#976e1d',
  colorsYellow900: '#644914',
  colorsYellow1000: '#32250a'
};
/** 主题色 */
var primaryColorVar = {
  sysColorPrimary50: '#fff5f4',
  sysColorPrimary100: '#feebe8',
  sysColorPrimary200: '#fdd6d1',
  sysColorPrimary300: '#fbada3',
  sysColorPrimary400: '#fa8474',
  sysColorPrimary500: '#f85b46',
  sysColorPrimary600: '#f63218',
  sysColorPrimary700: '#c52813',
  sysColorPrimary800: '#941e0e',
  sysColorPrimary900: '#62140a',
  sysColorPrimary1000: '#310a05'
};
/** 文本变量 */
var textColorVar = {
  sysTextPrimary: baseColorsVar.colorsNeutral1000,
  sysTextSecondary: baseColorsVar.colorsNeutral700,
  sysTextTertiary: baseColorsVar.colorsNeutral500,
  sysTextAnti: '#FFFFFF'
};
/** 边框色变量 */
var outlineVar = {
  sysSurfaceOutline300: baseColorsVar.colorsGray300,
  sysSurfaceOutline200: '#DCDEE0',
  sysSurfaceOutline100: baseColorsVar.colorsGray100,
  sysSurfaceOutline006: 'rgba(0, 0, 0, 0.06)'
};
/** 填充色变量 */
var fillVar = {
  sysSurfaceSurface300: baseColorsVar.colorsGray200,
  sysSurfaceSurface200: '#EBECEE',
  sysSurfaceSurface100: baseColorsVar.colorsGray50,
  sysSurfaceSurfaceWhite: '#FFFFFF'
};
/** 功能色 */
var stateVar = {
  sysStateInfo: baseColorsVar.colorsSkyBlue600,
  sysStateSuccess: baseColorsVar.colorsGreen600,
  sysStateWarning: baseColorsVar.colorsYellow600,
  sysStateError: baseColorsVar.colorsRed600,
  sysStateLink: baseColorsVar.colorsBlue600
};
var tokenVar = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, baseColorsVar), primaryColorVar), textColorVar), outlineVar), fillVar), stateVar), {}, {
  '1': '#FFFFFF',
  '11': '#000000',
  sysBackgroundWhite: '#FFFFFF',
  cNewgray06: '#FFFFFF',
  sysColorSecondary300: '#C4C7CC',
  sysBackgroundMain: '#F9F9F9',
  colorsAlphaWhite004: 'rgba(255, 255, 255, 0.04)',
  sysColorSecondary200: '#E1E3E6',
  sysColorSecondary100: '#F0F1F2',
  sysMoreAppAppSkyblue: '#53B2EE',
  sysMoreAppAppGreen: '#59C969',
  sysMoreAppAppPurple: '#995AF7',
  sysMoreAppAppRed: '#F85B46',
  sysMoreAppAppBlue: '#4687F7',
  sysSurfaceSurface100: '#F8F8F9',
  sysStateError100: '#FEEBE8',
  sysStateError300: '#FBADA3',
  sysStateError700: '#C52813',
  sysStateWarning100: '#FFF8EA',
  sysStateWarning300: '#FDE2AD',
  sysStateWarning700: '#C99227',
  sysStateSuccess100: '#EAF8EC',
  sysStateSuccess300: '#ACE4B4',
  sysStateSuccess700: '#269636',
  sysSurfaceOutline200: '#DCDEE0',
  colorsAlphaBlack064: 'rgba(0, 0, 0, 0.64)',
  primary: '#1B1B1B',
  characterDisabledPlaceholder25: 'rgba(0, 0, 0, 0.25)',
  comTooltipBgcolor: 'rgba(0, 0, 0, 0.92)',
  colorsAlphaWhite092: 'rgba(255, 255, 255, 0.92)',
  colorsBaseWhite: '#FFFFFF',
  sysColorSecondary500: '#898E99',
  comColorTabCurrentOnTab: '#F63218',
  sysColorSecondary800: '#40444D',
  sysSurfaceSurfaceRed: '#FFF5F4',
  colorsAlphaBlack032: 'rgba(0, 0, 0, 0.32)',
  sysColorSecondary900: '#2B2E33',
  colorsAlphaWhite032: 'rgba(255, 255, 255, 0.32)',
  colorsAlphaBlack092: 'rgba(0, 0, 0, 0.92)',
  sysColorSecondary400: '#A6AAB3',
  sysSurfaceOutline300: '#C4C7CC',
  colorBgMask: 'rgba(0, 0, 0, 0.64)',
  sysSurfaceSurface1: '#FFFFFF',
  sysSurfaceOutline2: 'rgba(0, 0, 0, 0.04)',
  sysSurfaceSurface2: '#F9F9F9',
  sysColorSecondary50: '#F8F8F9'
});

/** 获取kumaConfig配置 */
export var getKumaConfig = function getKumaConfig() {
  var prefix = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  var obj = {};
  Object.keys(tokenVar).forEach(function (e) {
    obj[_kebabCase(e)] = // @ts-ignore
    "var(".concat(prefix ? "--".concat(_kebabCase(prefix)) : "-", "-").concat(_kebabCase(e), ", ").concat(tokenVar[e], ")");
    obj[e] = // @ts-ignore
    "var(--".concat(e, ", ").concat(tokenVar[e], ")");
  });
  return obj;
};
export default tokenVar;
