// @jd/ant-theme@0.0.1-alpha.1/dist/Input/demo/special.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import styles from "../../Button/demo/index.module.less";
import { ErrorWrapper } from "../../components";
import { CopyOutlined, QuestionCircleOutlined, SearchOutlined, UserOutlined } from '@ant-design/icons';
import { Button, Input, InputNumber, message } from "../..";
import { useInterval } from 'ahooks';
import copy from 'copy-to-clipboard';
import React, { useRef, useState } from 'react';
import { runes } from 'runes2';
import EyesCloseIcon from "./eyes-close-icon";
import EyesIcon from "./eyes-icon";
var TextArea = Input.TextArea;

/**
 * 倒计时
 */
var TimeCount = function TimeCount() {
  var _useState = useState(0),
    _useState2 = _slicedToArray(_useState, 2),
    count = _useState2[0],
    setCount = _useState2[1];
  useInterval(function () {
    setCount(count - 1);
  }, count <= 0 ? undefined : 1000);
  return /*#__PURE__*/React.createElement("span", {
    className: "cursor-pointer",
    onClick: function onClick() {
      if (count === 0) {
        setCount(60);
      }
    }
  }, count === 0 ? '获取验证码' : "".concat(count, "s"));
};
/**
 * 特殊
 */
export default function () {
  var ref = useRef();
  var _message$useMessage = message.useMessage(),
    _message$useMessage2 = _slicedToArray(_message$useMessage, 2),
    messageApi = _message$useMessage2[0],
    contextHolder = _message$useMessage2[1];
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, contextHolder, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("table", {
    className: "text-sub text-left ".concat(styles.table)
  }, /*#__PURE__*/React.createElement("thead", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "\u7EC4\u5408icon"), /*#__PURE__*/React.createElement("td", null, "\u6570\u91CF\u9650\u5236"), /*#__PURE__*/React.createElement("td", null, "\u56FA\u5B9A\u524D\u7F00"), /*#__PURE__*/React.createElement("td", null, "\u5C3E\u90E8\u6309\u94AE"))), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("div", {
    className: "flex flex-col gap-16"
  }, /*#__PURE__*/React.createElement(Input, {
    prefix: /*#__PURE__*/React.createElement(UserOutlined, null),
    placeholder: "\u8BF7\u8F93\u5165"
  }), /*#__PURE__*/React.createElement(Input, {
    suffix: /*#__PURE__*/React.createElement(SearchOutlined, null),
    placeholder: "\u8BF7\u8F93\u5165"
  }))), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    count: {
      show: true,
      max: 6,
      strategy: function strategy(txt) {
        return runes(txt).length;
      },
      exceedFormatter: function exceedFormatter(txt, _ref) {
        var max = _ref.max;
        return runes(txt).slice(0, max).join('');
      }
    },
    defaultValue: "\uD83D\uDD25 antd"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("div", {
    className: "flex gap-6"
  }, /*#__PURE__*/React.createElement(Input, {
    addonBefore: "http://",
    placeholder: "\u8BF7\u8F93\u5165",
    suffix: /*#__PURE__*/React.createElement(QuestionCircleOutlined, null)
  }), /*#__PURE__*/React.createElement(Button, null, "\u6309\u94AE\u6587\u6848"))), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("div", {
    className: "flex flex-col gap-16"
  }, /*#__PURE__*/React.createElement(Input, {
    placeholder: "\u8BF7\u8F93\u5165",
    addonAfter: /*#__PURE__*/React.createElement(TimeCount, null)
  }), /*#__PURE__*/React.createElement(Input
  // @ts-ignore
  , {
    ref: ref,
    placeholder: "\u8BF7\u8F93\u5165",
    defaultValue: 'www.domain.com',
    addonAfter: /*#__PURE__*/React.createElement("div", {
      className: "flex items-center gap-6 cursor-pointer select-none",
      onClick: function onClick() {
        var _ref$current;
        // @ts-ignore
        copy((_ref$current = ref.current) === null || _ref$current === void 0 || (_ref$current = _ref$current.input) === null || _ref$current === void 0 ? void 0 : _ref$current.value);
        messageApi.open({
          type: 'success',
          content: '复制成功'
        });
      }
    }, /*#__PURE__*/React.createElement(CopyOutlined, {
      className: "text-10"
    }), /*#__PURE__*/React.createElement("span", null, "\u590D\u5236"))
  })))))), /*#__PURE__*/React.createElement("table", {
    className: "text-sub text-left ".concat(styles.table)
  }, /*#__PURE__*/React.createElement("thead", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "\u5BC6\u7801\u8F93\u5165"), /*#__PURE__*/React.createElement("td", null, "\u6587\u672C\u6846\u8F93\u5165"))), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input.Password, {
    placeholder: "input password",
    iconRender: function iconRender(visible) {
      return visible ? /*#__PURE__*/React.createElement(EyesIcon, null) : /*#__PURE__*/React.createElement(EyesCloseIcon, null);
    }
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(TextArea, {
    rows: 6,
    className: "w-480",
    placeholder: "\u8F93\u5165\u6587\u5B57\u63CF\u8FF0...",
    allowClear: true,
    count: {
      show: true,
      max: 1000,
      strategy: function strategy(txt) {
        return runes(txt).length;
      },
      exceedFormatter: function exceedFormatter(txt, _ref2) {
        var max = _ref2.max;
        return runes(txt).slice(0, max).join('');
      }
    }
  }))))), /*#__PURE__*/React.createElement("table", {
    className: "text-sub text-left ".concat(styles.table)
  }, /*#__PURE__*/React.createElement("thead", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "\u6570\u91CF\u9009\u62E9\u5668"))), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(InputNumber, {
    min: 1,
    max: 10,
    defaultValue: 3
  })))))));
}
