// @jd/react-custom-theme@0.0.1-alpha.4/dist/es/defaultToken-C2dt-V1j.js
var f = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function M(r) {
  return r && r.__esModule && Object.prototype.hasOwnProperty.call(r, "default") ? r.default : r;
}
var L = typeof f == "object" && f && f.Object === Object && f, U = L, D = U, Y = typeof self == "object" && self && self.Object === Object && self, z = D || Y || Function("return this")(), Z = z, V = Z, H = V.Symbol, d = H, b = d, A = Object.prototype, J = A.hasOwnProperty, K = A.toString, n = b ? b.toStringTag : void 0;
function q(r) {
  var o = J.call(r, n), e = r[n];
  try {
    r[n] = void 0;
    var a = !0;
  } catch {
  }
  var c = K.call(r);
  return a && (o ? r[n] = e : delete r[n]), c;
}
var Q = q, X = Object.prototype, r0 = X.toString;
function o0(r) {
  return r0.call(r);
}
var e0 = o0, y = d, s0 = Q, a0 = e0, c0 = "[object Null]", u0 = "[object Undefined]", g = y ? y.toStringTag : void 0;
function l0(r) {
  return r == null ? r === void 0 ? u0 : c0 : g && g in Object(r) ? s0(r) : a0(r);
}
var n0 = l0;
function f0(r) {
  return r != null && typeof r == "object";
}
var t0 = f0, d0 = Array.isArray, i0 = d0;
function b0(r, o, e, a) {
  var c = -1, l = r == null ? 0 : r.length;
  for (a && l && (e = r[++c]); ++c < l; )
    e = o(e, r[c], c, r);
  return e;
}
var y0 = b0;
function g0(r) {
  return function(o) {
    return r == null ? void 0 : r[o];
  };
}
var x0 = g0, S0 = x0, p0 = {
  // Latin-1 Supplement block.
  À: "A",
  Á: "A",
  Â: "A",
  Ã: "A",
  Ä: "A",
  Å: "A",
  à: "a",
  á: "a",
  â: "a",
  ã: "a",
  ä: "a",
  å: "a",
  Ç: "C",
  ç: "c",
  Ð: "D",
  ð: "d",
  È: "E",
  É: "E",
  Ê: "E",
  Ë: "E",
  è: "e",
  é: "e",
  ê: "e",
  ë: "e",
  Ì: "I",
  Í: "I",
  Î: "I",
  Ï: "I",
  ì: "i",
  í: "i",
  î: "i",
  ï: "i",
  Ñ: "N",
  ñ: "n",
  Ò: "O",
  Ó: "O",
  Ô: "O",
  Õ: "O",
  Ö: "O",
  Ø: "O",
  ò: "o",
  ó: "o",
  ô: "o",
  õ: "o",
  ö: "o",
  ø: "o",
  Ù: "U",
  Ú: "U",
  Û: "U",
  Ü: "U",
  ù: "u",
  ú: "u",
  û: "u",
  ü: "u",
  Ý: "Y",
  ý: "y",
  ÿ: "y",
  Æ: "Ae",
  æ: "ae",
  Þ: "Th",
  þ: "th",
  ß: "ss",
  // Latin Extended-A block.
  Ā: "A",
  Ă: "A",
  Ą: "A",
  ā: "a",
  ă: "a",
  ą: "a",
  Ć: "C",
  Ĉ: "C",
  Ċ: "C",
  Č: "C",
  ć: "c",
  ĉ: "c",
  ċ: "c",
  č: "c",
  Ď: "D",
  Đ: "D",
  ď: "d",
  đ: "d",
  Ē: "E",
  Ĕ: "E",
  Ė: "E",
  Ę: "E",
  Ě: "E",
  ē: "e",
  ĕ: "e",
  ė: "e",
  ę: "e",
  ě: "e",
  Ĝ: "G",
  Ğ: "G",
  Ġ: "G",
  Ģ: "G",
  ĝ: "g",
  ğ: "g",
  ġ: "g",
  ģ: "g",
  Ĥ: "H",
  Ħ: "H",
  ĥ: "h",
  ħ: "h",
  Ĩ: "I",
  Ī: "I",
  Ĭ: "I",
  Į: "I",
  İ: "I",
  ĩ: "i",
  ī: "i",
  ĭ: "i",
  į: "i",
  ı: "i",
  Ĵ: "J",
  ĵ: "j",
  Ķ: "K",
  ķ: "k",
  ĸ: "k",
  Ĺ: "L",
  Ļ: "L",
  Ľ: "L",
  Ŀ: "L",
  Ł: "L",
  ĺ: "l",
  ļ: "l",
  ľ: "l",
  ŀ: "l",
  ł: "l",
  Ń: "N",
  Ņ: "N",
  Ň: "N",
  Ŋ: "N",
  ń: "n",
  ņ: "n",
  ň: "n",
  ŋ: "n",
  Ō: "O",
  Ŏ: "O",
  Ő: "O",
  ō: "o",
  ŏ: "o",
  ő: "o",
  Ŕ: "R",
  Ŗ: "R",
  Ř: "R",
  ŕ: "r",
  ŗ: "r",
  ř: "r",
  Ś: "S",
  Ŝ: "S",
  Ş: "S",
  Š: "S",
  ś: "s",
  ŝ: "s",
  ş: "s",
  š: "s",
  Ţ: "T",
  Ť: "T",
  Ŧ: "T",
  ţ: "t",
  ť: "t",
  ŧ: "t",
  Ũ: "U",
  Ū: "U",
  Ŭ: "U",
  Ů: "U",
  Ű: "U",
  Ų: "U",
  ũ: "u",
  ū: "u",
  ŭ: "u",
  ů: "u",
  ű: "u",
  ų: "u",
  Ŵ: "W",
  ŵ: "w",
  Ŷ: "Y",
  ŷ: "y",
  Ÿ: "Y",
  Ź: "Z",
  Ż: "Z",
  Ž: "Z",
  ź: "z",
  ż: "z",
  ž: "z",
  Ĳ: "IJ",
  ĳ: "ij",
  Œ: "Oe",
  œ: "oe",
  ŉ: "'n",
  ſ: "s"
}, F0 = S0(p0), C0 = F0;
function v0(r, o) {
  for (var e = -1, a = r == null ? 0 : r.length, c = Array(a); ++e < a; )
    c[e] = o(r[e], e, r);
  return c;
}
var O0 = v0, m0 = n0, A0 = t0, h0 = "[object Symbol]";
function k0(r) {
  return typeof r == "symbol" || A0(r) && m0(r) == h0;
}
var B0 = k0, x = d, R0 = O0, P0 = i0, E0 = B0, $0 = 1 / 0, S = x ? x.prototype : void 0, p = S ? S.toString : void 0;
function h(r) {
  if (typeof r == "string")
    return r;
  if (P0(r))
    return R0(r, h) + "";
  if (E0(r))
    return p ? p.call(r) : "";
  var o = r + "";
  return o == "0" && 1 / r == -$0 ? "-0" : o;
}
var T0 = h, G0 = T0;
function j0(r) {
  return r == null ? "" : G0(r);
}
var k = j0, w0 = C0, _0 = k, N0 = /[\xc0-\xd6\xd8-\xf6\xf8-\xff\u0100-\u017f]/g, I0 = "\\u0300-\\u036f", W0 = "\\ufe20-\\ufe2f", M0 = "\\u20d0-\\u20ff", L0 = I0 + W0 + M0, U0 = "[" + L0 + "]", D0 = RegExp(U0, "g");
function Y0(r) {
  return r = _0(r), r && r.replace(N0, w0).replace(D0, "");
}
var z0 = Y0, Z0 = /[^\x00-\x2f\x3a-\x40\x5b-\x60\x7b-\x7f]+/g;
function V0(r) {
  return r.match(Z0) || [];
}
var H0 = V0, J0 = /[a-z][A-Z]|[A-Z]{2}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]/;
function K0(r) {
  return J0.test(r);
}
var q0 = K0, B = "\\ud800-\\udfff", Q0 = "\\u0300-\\u036f", X0 = "\\ufe20-\\ufe2f", rr = "\\u20d0-\\u20ff", or = Q0 + X0 + rr, R = "\\u2700-\\u27bf", P = "a-z\\xdf-\\xf6\\xf8-\\xff", er = "\\xac\\xb1\\xd7\\xf7", sr = "\\x00-\\x2f\\x3a-\\x40\\x5b-\\x60\\x7b-\\xbf", ar = "\\u2000-\\u206f", cr = " \\t\\x0b\\f\\xa0\\ufeff\\n\\r\\u2028\\u2029\\u1680\\u180e\\u2000\\u2001\\u2002\\u2003\\u2004\\u2005\\u2006\\u2007\\u2008\\u2009\\u200a\\u202f\\u205f\\u3000", E = "A-Z\\xc0-\\xd6\\xd8-\\xde", ur = "\\ufe0e\\ufe0f", $ = er + sr + ar + cr, T = "['’]", F = "[" + $ + "]", lr = "[" + or + "]", G = "\\d+", nr = "[" + R + "]", j = "[" + P + "]", w = "[^" + B + $ + G + R + P + E + "]", fr = "\\ud83c[\\udffb-\\udfff]", tr = "(?:" + lr + "|" + fr + ")", dr = "[^" + B + "]", _ = "(?:\\ud83c[\\udde6-\\uddff]){2}", N = "[\\ud800-\\udbff][\\udc00-\\udfff]", u = "[" + E + "]", ir = "\\u200d", C = "(?:" + j + "|" + w + ")", br = "(?:" + u + "|" + w + ")", v = "(?:" + T + "(?:d|ll|m|re|s|t|ve))?", O = "(?:" + T + "(?:D|LL|M|RE|S|T|VE))?", I = tr + "?", W = "[" + ur + "]?", yr = "(?:" + ir + "(?:" + [dr, _, N].join("|") + ")" + W + I + ")*", gr = "\\d*(?:1st|2nd|3rd|(?![123])\\dth)(?=\\b|[A-Z_])", xr = "\\d*(?:1ST|2ND|3RD|(?![123])\\dTH)(?=\\b|[a-z_])", Sr = W + I + yr, pr = "(?:" + [nr, _, N].join("|") + ")" + Sr, Fr = RegExp([
  u + "?" + j + "+" + v + "(?=" + [F, u, "$"].join("|") + ")",
  br + "+" + O + "(?=" + [F, u + C, "$"].join("|") + ")",
  u + "?" + C + "+" + v,
  u + "+" + O,
  xr,
  gr,
  G,
  pr
].join("|"), "g");
function Cr(r) {
  return r.match(Fr) || [];
}
var vr = Cr, Or = H0, mr = q0, Ar = k, hr = vr;
function kr(r, o, e) {
  return r = Ar(r), o = e ? void 0 : o, o === void 0 ? mr(r) ? hr(r) : Or(r) : r.match(o) || [];
}
var Br = kr, Rr = y0, Pr = z0, Er = Br, $r = "['’]", Tr = RegExp($r, "g");
function Gr(r) {
  return function(o) {
    return Rr(Er(Pr(o).replace(Tr, "")), r, "");
  };
}
var jr = Gr, wr = jr, _r = wr(function(r, o, e) {
  return r + (e ? "-" : "") + o.toLowerCase();
}), Nr = _r;
const t = /* @__PURE__ */ M(Nr), s = {
  colorsNeutral100: "#f3f3f3",
  colorsNeutral200: "#e7e7e7",
  colorsNeutral300: "#cecece",
  colorsNeutral400: "#B6B6B6",
  colorsNeutral500: "#9d9d9d",
  colorsNeutral600: "#858585",
  colorsNeutral700: "#6a6a6a",
  colorsNeutral800: "#505050",
  colorsNeutral900: "#353535",
  colorsNeutral1000: "#1b1b1b",
  colorsGray50: "#f8f8f9",
  colorsGray100: "#f0f1f2",
  colorsGray200: "#e1e3e6",
  colorsGray300: "#c4c7cc",
  colorsGray400: "#a6aab3",
  colorsGray500: "#898e99",
  colorsGray600: "#6b7280",
  colorsGray700: "#565b66",
  colorsGray800: "#40444d",
  colorsGray900: "#2b2e33",
  colorsGray1000: "#15171a",
  colorsGreen100: "#eaf8ec",
  colorsGreen200: "#d5f2da",
  colorsGreen300: "#ace4b4",
  colorsGreen400: "#82d78f",
  colorsGreen500: "#59c969",
  colorsGreen600: "#2fbc44",
  colorsGreen700: "#269636",
  colorsGreen800: "#1c7129",
  colorsGreen900: "#134b1b",
  colorsGreen1000: "#09260e",
  colorsCyan100: "#e8fafd",
  colorsCyan200: "#d1f5fb",
  colorsCyan300: "#a3ebf8",
  colorsCyan400: "#76e1f4",
  colorsCyan500: "#48d7f1",
  colorsCyan600: "#1acded",
  colorsCyan700: "#15a4be",
  colorsCyan800: "#107b8e",
  colorsCyan900: "#0a525f",
  colorsCyan1000: "#05292f",
  colorsSkyBlue100: "#eaf5fd",
  colorsSkyBlue200: "#d4ecfb",
  colorsSkyBlue300: "#a9d9f7",
  colorsSkyBlue400: "#7ec5f2",
  colorsSkyBlue500: "#53b2ee",
  colorsSkyBlue600: "#289fea",
  colorsSkyBlue700: "#207fbb",
  colorsSkyBlue800: "#185f8c",
  colorsSkyBlue900: "#10405e",
  colorsSkyBlue1000: "#08202f",
  colorsBlue100: "#e8f0fe",
  colorsBlue200: "#d1e1fd",
  colorsBlue300: "#a3c3fb",
  colorsBlue400: "#74a5f9",
  colorsBlue500: "#4687f7",
  colorsBlue600: "#1869f5",
  colorsBlue700: "#1354c4",
  colorsBlue800: "#0e3f93",
  colorsBlue900: "#0a2a62",
  colorsBlue1000: "#051531",
  colorsIndigo100: "#edeffd",
  colorsIndigo200: "#dbdffa",
  colorsIndigo300: "#b7bff6",
  colorsIndigo400: "#929ff1",
  colorsIndigo500: "#6e7fed",
  colorsIndigo600: "#4a5fe8",
  colorsIndigo700: "#3b4cba",
  colorsIndigo800: "#2c398b",
  colorsIndigo900: "#1e265d",
  colorsIndigo1000: "#0f132e",
  colorsPurple100: "#f2eafe",
  colorsPurple200: "#e6d6fd",
  colorsPurple300: "#ccadfb",
  colorsPurple400: "#b383f9",
  colorsPurple500: "#995af7",
  colorsPurple600: "#8031f5",
  colorsPurple700: "#6627c4",
  colorsPurple800: "#4d1d93",
  colorsPurple900: "#331462",
  colorsPurple1000: "#1a0a31",
  colorsFuchsia100: "#f9ecfb",
  colorsFuchsia200: "#f3d8f8",
  colorsFuchsia300: "#e6b1f1",
  colorsFuchsia400: "#da8be9",
  colorsFuchsia500: "#cd64e2",
  colorsFuchsia600: "#c13ddb",
  colorsFuchsia700: "#9a31af",
  colorsFuchsia800: "#742583",
  colorsFuchsia900: "#4d1858",
  colorsFuchsia1000: "#270c2c",
  colorsPink100: "#feebf3",
  colorsPink200: "#fdd7e8",
  colorsPink300: "#fcafd0",
  colorsPink400: "#fa86b9",
  colorsPink500: "#f95ea1",
  colorsPink600: "#f7368a",
  colorsPink700: "#c62b6e",
  colorsPink800: "#942053",
  colorsPink900: "#631637",
  colorsPink1000: "#310b1c",
  colorsRed50: "#fff5f4",
  colorsRed100: "#feebe8",
  colorsRed200: "#fdd6d1",
  colorsRed300: "#fbada3",
  colorsRed400: "#fa8474",
  colorsRed500: "#f85b46",
  colorsRed600: "#f63218",
  colorsRed700: "#c52813",
  colorsRed800: "#941e0e",
  colorsRed900: "#62140a",
  colorsRed1000: "#310a05",
  colorsOrange100: "#feeee8",
  colorsOrange200: "#fcdcd1",
  colorsOrange300: "#fabaa3",
  colorsOrange400: "#f79774",
  colorsOrange500: "#f95ea1",
  colorsOrange600: "#f25218",
  colorsOrange700: "#c24213",
  colorsOrange800: "#91310e",
  colorsOrange900: "#61210a",
  colorsOrange1000: "#301005",
  colorsYellow100: "#fff8ea",
  colorsYellow200: "#fef1d6",
  colorsYellow300: "#fde2ad",
  colorsYellow400: "#fdd483",
  colorsYellow500: "#fcc55a",
  colorsYellow600: "#fbb731",
  colorsYellow700: "#c99227",
  colorsYellow800: "#976e1d",
  colorsYellow900: "#644914",
  colorsYellow1000: "#32250a"
}, Ir = {
  sysColorPrimary50: "#fff5f4",
  sysColorPrimary100: "#feebe8",
  sysColorPrimary200: "#fdd6d1",
  sysColorPrimary300: "#fbada3",
  sysColorPrimary400: "#fa8474",
  sysColorPrimary500: "#f85b46",
  sysColorPrimary600: "#f63218",
  sysColorPrimary700: "#c52813",
  sysColorPrimary800: "#941e0e",
  sysColorPrimary900: "#62140a",
  sysColorPrimary1000: "#310a05"
}, Wr = {
  sysTextPrimary: s.colorsNeutral1000,
  sysTextSecondary: s.colorsNeutral700,
  sysTextTertiary: s.colorsNeutral500,
  sysTextAnti: "#FFFFFF"
}, Mr = {
  sysSurfaceOutline300: s.colorsGray300,
  sysSurfaceOutline200: "#DCDEE0",
  sysSurfaceOutline100: s.colorsGray100,
  sysSurfaceOutline006: "rgba(0, 0, 0, 0.06)"
}, Lr = {
  sysSurfaceSurface300: s.colorsGray200,
  sysSurfaceSurface200: "#EBECEE",
  sysSurfaceSurface100: s.colorsGray50,
  sysSurfaceSurfaceWhite: "#FFFFFF"
}, Ur = {
  sysStateInfo: s.colorsSkyBlue600,
  sysStateSuccess: s.colorsGreen600,
  sysStateWarning: s.colorsYellow600,
  sysStateError: s.colorsRed600,
  sysStateLink: s.colorsBlue600
}, m = {
  ...s,
  ...Ir,
  ...Wr,
  ...Mr,
  ...Lr,
  ...Ur,
  1: "#FFFFFF",
  11: "#000000",
  sysBackgroundWhite: "#FFFFFF",
  cNewgray06: "#FFFFFF",
  sysColorSecondary300: "#C4C7CC",
  sysBackgroundMain: "#F9F9F9",
  colorsAlphaWhite004: "rgba(255, 255, 255, 0.04)",
  sysColorSecondary200: "#E1E3E6",
  sysColorSecondary100: "#F0F1F2",
  sysMoreAppAppSkyblue: "#53B2EE",
  sysMoreAppAppGreen: "#59C969",
  sysMoreAppAppPurple: "#995AF7",
  sysMoreAppAppRed: "#F85B46",
  sysMoreAppAppBlue: "#4687F7",
  sysSurfaceSurface100: "#F8F8F9",
  sysStateError100: "#FEEBE8",
  sysStateError300: "#FBADA3",
  sysStateError700: "#C52813",
  sysStateWarning100: "#FFF8EA",
  sysStateWarning300: "#FDE2AD",
  sysStateWarning700: "#C99227",
  sysStateSuccess100: "#EAF8EC",
  sysStateSuccess300: "#ACE4B4",
  sysStateSuccess700: "#269636",
  sysSurfaceOutline200: "#DCDEE0",
  colorsAlphaBlack064: "rgba(0, 0, 0, 0.64)",
  primary: "#1B1B1B",
  characterDisabledPlaceholder25: "rgba(0, 0, 0, 0.25)",
  comTooltipBgcolor: "rgba(0, 0, 0, 0.92)",
  colorsAlphaWhite092: "rgba(255, 255, 255, 0.92)",
  colorsBaseWhite: "#FFFFFF",
  sysColorSecondary500: "#898E99",
  comColorTabCurrentOnTab: "#F63218",
  sysColorSecondary800: "#40444D",
  sysSurfaceSurfaceRed: "#FFF5F4",
  colorsAlphaBlack032: "rgba(0, 0, 0, 0.32)",
  sysColorSecondary900: "#2B2E33",
  colorsAlphaWhite032: "rgba(255, 255, 255, 0.32)",
  colorsAlphaBlack092: "rgba(0, 0, 0, 0.92)",
  sysColorSecondary400: "#A6AAB3",
  sysSurfaceOutline300: "#C4C7CC",
  colorBgMask: "rgba(0, 0, 0, 0.64)",
  sysSurfaceSurface1: "#FFFFFF",
  sysSurfaceOutline2: "rgba(0, 0, 0, 0.04)",
  sysSurfaceSurface2: "#F9F9F9",
  sysColorSecondary50: "#F8F8F9"
};
function Dr(r = "", o = m) {
  const e = {};
  return Object.keys(o).forEach((a) => {
    const c = `${r ? `--${t(r)}` : "-"}-${t(a)}`, l = `--${a}`, i = m[a];
    e[t(a)] = `var(${c}, ${i})`, e[a] = `var(${l}, ${i})`;
  }), e;
}
export {
  n0 as _,
  Z as a,
  U as b,
  i0 as c,
  Dr as d,
  M as g,
  t0 as i,
  m as t
};
