// @jd/react-custom-theme@0.0.1-alpha.4/dist/es/index.js
import { _ as A, a as y, i as b, b as ge, c as Z, g as $e, t as Q, d as de } from "./defaultToken-C2dt-V1j.js";
import { useEffect as ye } from "react";
function be() {
  this.__data__ = [], this.size = 0;
}
var Oe = be;
function Ce(e, r) {
  return e === r || e !== e && r !== r;
}
var j = Ce, Te = j;
function Ae(e, r) {
  for (var t = e.length; t--; )
    if (Te(e[t][0], r))
      return t;
  return -1;
}
var m = Ae, je = m, me = Array.prototype, Pe = me.splice;
function xe(e) {
  var r = this.__data__, t = je(r, e);
  if (t < 0)
    return !1;
  var a = r.length - 1;
  return t == a ? r.pop() : Pe.call(r, t, 1), --this.size, !0;
}
var Ie = xe, Se = m;
function we(e) {
  var r = this.__data__, t = Se(r, e);
  return t < 0 ? void 0 : r[t][1];
}
var Me = we, De = m;
function Ee(e) {
  return De(this.__data__, e) > -1;
}
var Le = Ee, Fe = m;
function Ge(e, r) {
  var t = this.__data__, a = Fe(t, e);
  return a < 0 ? (++this.size, t.push([e, r])) : t[a][1] = r, this;
}
var He = Ge, Be = Oe, Ue = Ie, Ne = Me, ze = Le, Re = He;
function _(e) {
  var r = -1, t = e == null ? 0 : e.length;
  for (this.clear(); ++r < t; ) {
    var a = e[r];
    this.set(a[0], a[1]);
  }
}
_.prototype.clear = Be;
_.prototype.delete = Ue;
_.prototype.get = Ne;
_.prototype.has = ze;
_.prototype.set = Re;
var P = _, Ve = P;
function Ke() {
  this.__data__ = new Ve(), this.size = 0;
}
var qe = Ke;
function ke(e) {
  var r = this.__data__, t = r.delete(e);
  return this.size = r.size, t;
}
var Je = ke;
function Xe(e) {
  return this.__data__.get(e);
}
var We = Xe;
function Ye(e) {
  return this.__data__.has(e);
}
var Ze = Ye;
function Qe(e) {
  var r = typeof e;
  return e != null && (r == "object" || r == "function");
}
var p = Qe, er = A, rr = p, tr = "[object AsyncFunction]", ar = "[object Function]", nr = "[object GeneratorFunction]", ir = "[object Proxy]";
function sr(e) {
  if (!rr(e))
    return !1;
  var r = er(e);
  return r == ar || r == nr || r == tr || r == ir;
}
var M = sr, or = y, cr = or["__core-js_shared__"], ur = cr, w = ur, G = function() {
  var e = /[^.]+$/.exec(w && w.keys && w.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function fr(e) {
  return !!G && G in e;
}
var vr = fr, lr = Function.prototype, pr = lr.toString;
function hr(e) {
  if (e != null) {
    try {
      return pr.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var _r = hr, gr = M, $r = vr, dr = p, yr = _r, br = /[\\^$.*+?()[\]{}|]/g, Or = /^\[object .+?Constructor\]$/, Cr = Function.prototype, Tr = Object.prototype, Ar = Cr.toString, jr = Tr.hasOwnProperty, mr = RegExp(
  "^" + Ar.call(jr).replace(br, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Pr(e) {
  if (!dr(e) || $r(e))
    return !1;
  var r = gr(e) ? mr : Or;
  return r.test(yr(e));
}
var xr = Pr;
function Ir(e, r) {
  return e == null ? void 0 : e[r];
}
var Sr = Ir, wr = xr, Mr = Sr;
function Dr(e, r) {
  var t = Mr(e, r);
  return wr(t) ? t : void 0;
}
var D = Dr, Er = D, Lr = y, Fr = Er(Lr, "Map"), ee = Fr, Gr = D, Hr = Gr(Object, "create"), x = Hr, H = x;
function Br() {
  this.__data__ = H ? H(null) : {}, this.size = 0;
}
var Ur = Br;
function Nr(e) {
  var r = this.has(e) && delete this.__data__[e];
  return this.size -= r ? 1 : 0, r;
}
var zr = Nr, Rr = x, Vr = "__lodash_hash_undefined__", Kr = Object.prototype, qr = Kr.hasOwnProperty;
function kr(e) {
  var r = this.__data__;
  if (Rr) {
    var t = r[e];
    return t === Vr ? void 0 : t;
  }
  return qr.call(r, e) ? r[e] : void 0;
}
var Jr = kr, Xr = x, Wr = Object.prototype, Yr = Wr.hasOwnProperty;
function Zr(e) {
  var r = this.__data__;
  return Xr ? r[e] !== void 0 : Yr.call(r, e);
}
var Qr = Zr, et = x, rt = "__lodash_hash_undefined__";
function tt(e, r) {
  var t = this.__data__;
  return this.size += this.has(e) ? 0 : 1, t[e] = et && r === void 0 ? rt : r, this;
}
var at = tt, nt = Ur, it = zr, st = Jr, ot = Qr, ct = at;
function g(e) {
  var r = -1, t = e == null ? 0 : e.length;
  for (this.clear(); ++r < t; ) {
    var a = e[r];
    this.set(a[0], a[1]);
  }
}
g.prototype.clear = nt;
g.prototype.delete = it;
g.prototype.get = st;
g.prototype.has = ot;
g.prototype.set = ct;
var ut = g, B = ut, ft = P, vt = ee;
function lt() {
  this.size = 0, this.__data__ = {
    hash: new B(),
    map: new (vt || ft)(),
    string: new B()
  };
}
var pt = lt;
function ht(e) {
  var r = typeof e;
  return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? e !== "__proto__" : e === null;
}
var _t = ht, gt = _t;
function $t(e, r) {
  var t = e.__data__;
  return gt(r) ? t[typeof r == "string" ? "string" : "hash"] : t.map;
}
var I = $t, dt = I;
function yt(e) {
  var r = dt(this, e).delete(e);
  return this.size -= r ? 1 : 0, r;
}
var bt = yt, Ot = I;
function Ct(e) {
  return Ot(this, e).get(e);
}
var Tt = Ct, At = I;
function jt(e) {
  return At(this, e).has(e);
}
var mt = jt, Pt = I;
function xt(e, r) {
  var t = Pt(this, e), a = t.size;
  return t.set(e, r), this.size += t.size == a ? 0 : 1, this;
}
var It = xt, St = pt, wt = bt, Mt = Tt, Dt = mt, Et = It;
function $(e) {
  var r = -1, t = e == null ? 0 : e.length;
  for (this.clear(); ++r < t; ) {
    var a = e[r];
    this.set(a[0], a[1]);
  }
}
$.prototype.clear = St;
$.prototype.delete = wt;
$.prototype.get = Mt;
$.prototype.has = Dt;
$.prototype.set = Et;
var Lt = $, Ft = P, Gt = ee, Ht = Lt, Bt = 200;
function Ut(e, r) {
  var t = this.__data__;
  if (t instanceof Ft) {
    var a = t.__data__;
    if (!Gt || a.length < Bt - 1)
      return a.push([e, r]), this.size = ++t.size, this;
    t = this.__data__ = new Ht(a);
  }
  return t.set(e, r), this.size = t.size, this;
}
var Nt = Ut, zt = P, Rt = qe, Vt = Je, Kt = We, qt = Ze, kt = Nt;
function d(e) {
  var r = this.__data__ = new zt(e);
  this.size = r.size;
}
d.prototype.clear = Rt;
d.prototype.delete = Vt;
d.prototype.get = Kt;
d.prototype.has = qt;
d.prototype.set = kt;
var Jt = d, Xt = D, Wt = function() {
  try {
    var e = Xt(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), re = Wt, U = re;
function Yt(e, r, t) {
  r == "__proto__" && U ? U(e, r, {
    configurable: !0,
    enumerable: !0,
    value: t,
    writable: !0
  }) : e[r] = t;
}
var E = Yt, Zt = E, Qt = j;
function ea(e, r, t) {
  (t !== void 0 && !Qt(e[r], t) || t === void 0 && !(r in e)) && Zt(e, r, t);
}
var te = ea;
function ra(e) {
  return function(r, t, a) {
    for (var i = -1, s = Object(r), o = a(r), n = o.length; n--; ) {
      var c = o[e ? n : ++i];
      if (t(s[c], c, s) === !1)
        break;
    }
    return r;
  };
}
var ta = ra, aa = ta, na = aa(), ia = na, O = { exports: {} };
O.exports;
(function(e, r) {
  var t = y, a = r && !r.nodeType && r, i = a && !0 && e && !e.nodeType && e, s = i && i.exports === a, o = s ? t.Buffer : void 0, n = o ? o.allocUnsafe : void 0;
  function c(f, v) {
    if (v)
      return f.slice();
    var l = f.length, h = n ? n(l) : new f.constructor(l);
    return f.copy(h), h;
  }
  e.exports = c;
})(O, O.exports);
var sa = O.exports, oa = y, ca = oa.Uint8Array, ua = ca, N = ua;
function fa(e) {
  var r = new e.constructor(e.byteLength);
  return new N(r).set(new N(e)), r;
}
var va = fa, la = va;
function pa(e, r) {
  var t = r ? la(e.buffer) : e.buffer;
  return new e.constructor(t, e.byteOffset, e.length);
}
var ha = pa;
function _a(e, r) {
  var t = -1, a = e.length;
  for (r || (r = Array(a)); ++t < a; )
    r[t] = e[t];
  return r;
}
var ga = _a, $a = p, z = Object.create, da = /* @__PURE__ */ function() {
  function e() {
  }
  return function(r) {
    if (!$a(r))
      return {};
    if (z)
      return z(r);
    e.prototype = r;
    var t = new e();
    return e.prototype = void 0, t;
  };
}(), ya = da;
function ba(e, r) {
  return function(t) {
    return e(r(t));
  };
}
var Oa = ba, Ca = Oa, Ta = Ca(Object.getPrototypeOf, Object), ae = Ta, Aa = Object.prototype;
function ja(e) {
  var r = e && e.constructor, t = typeof r == "function" && r.prototype || Aa;
  return e === t;
}
var ne = ja, ma = ya, Pa = ae, xa = ne;
function Ia(e) {
  return typeof e.constructor == "function" && !xa(e) ? ma(Pa(e)) : {};
}
var Sa = Ia, wa = A, Ma = b, Da = "[object Arguments]";
function Ea(e) {
  return Ma(e) && wa(e) == Da;
}
var La = Ea, R = La, Fa = b, ie = Object.prototype, Ga = ie.hasOwnProperty, Ha = ie.propertyIsEnumerable, Ba = R(/* @__PURE__ */ function() {
  return arguments;
}()) ? R : function(e) {
  return Fa(e) && Ga.call(e, "callee") && !Ha.call(e, "callee");
}, se = Ba, Ua = 9007199254740991;
function Na(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Ua;
}
var oe = Na, za = M, Ra = oe;
function Va(e) {
  return e != null && Ra(e.length) && !za(e);
}
var L = Va, Ka = L, qa = b;
function ka(e) {
  return qa(e) && Ka(e);
}
var Ja = ka, C = { exports: {} };
function Xa() {
  return !1;
}
var Wa = Xa;
C.exports;
(function(e, r) {
  var t = y, a = Wa, i = r && !r.nodeType && r, s = i && !0 && e && !e.nodeType && e, o = s && s.exports === i, n = o ? t.Buffer : void 0, c = n ? n.isBuffer : void 0, f = c || a;
  e.exports = f;
})(C, C.exports);
var ce = C.exports, Ya = A, Za = ae, Qa = b, en = "[object Object]", rn = Function.prototype, tn = Object.prototype, ue = rn.toString, an = tn.hasOwnProperty, nn = ue.call(Object);
function sn(e) {
  if (!Qa(e) || Ya(e) != en)
    return !1;
  var r = Za(e);
  if (r === null)
    return !0;
  var t = an.call(r, "constructor") && r.constructor;
  return typeof t == "function" && t instanceof t && ue.call(t) == nn;
}
var on = sn, cn = A, un = oe, fn = b, vn = "[object Arguments]", ln = "[object Array]", pn = "[object Boolean]", hn = "[object Date]", _n = "[object Error]", gn = "[object Function]", $n = "[object Map]", dn = "[object Number]", yn = "[object Object]", bn = "[object RegExp]", On = "[object Set]", Cn = "[object String]", Tn = "[object WeakMap]", An = "[object ArrayBuffer]", jn = "[object DataView]", mn = "[object Float32Array]", Pn = "[object Float64Array]", xn = "[object Int8Array]", In = "[object Int16Array]", Sn = "[object Int32Array]", wn = "[object Uint8Array]", Mn = "[object Uint8ClampedArray]", Dn = "[object Uint16Array]", En = "[object Uint32Array]", u = {};
u[mn] = u[Pn] = u[xn] = u[In] = u[Sn] = u[wn] = u[Mn] = u[Dn] = u[En] = !0;
u[vn] = u[ln] = u[An] = u[pn] = u[jn] = u[hn] = u[_n] = u[gn] = u[$n] = u[dn] = u[yn] = u[bn] = u[On] = u[Cn] = u[Tn] = !1;
function Ln(e) {
  return fn(e) && un(e.length) && !!u[cn(e)];
}
var Fn = Ln;
function Gn(e) {
  return function(r) {
    return e(r);
  };
}
var Hn = Gn, T = { exports: {} };
T.exports;
(function(e, r) {
  var t = ge, a = r && !r.nodeType && r, i = a && !0 && e && !e.nodeType && e, s = i && i.exports === a, o = s && t.process, n = function() {
    try {
      var c = i && i.require && i.require("util").types;
      return c || o && o.binding && o.binding("util");
    } catch {
    }
  }();
  e.exports = n;
})(T, T.exports);
var Bn = T.exports, Un = Fn, Nn = Hn, V = Bn, K = V && V.isTypedArray, zn = K ? Nn(K) : Un, fe = zn;
function Rn(e, r) {
  if (!(r === "constructor" && typeof e[r] == "function") && r != "__proto__")
    return e[r];
}
var ve = Rn, Vn = E, Kn = j, qn = Object.prototype, kn = qn.hasOwnProperty;
function Jn(e, r, t) {
  var a = e[r];
  (!(kn.call(e, r) && Kn(a, t)) || t === void 0 && !(r in e)) && Vn(e, r, t);
}
var Xn = Jn, Wn = Xn, Yn = E;
function Zn(e, r, t, a) {
  var i = !t;
  t || (t = {});
  for (var s = -1, o = r.length; ++s < o; ) {
    var n = r[s], c = a ? a(t[n], e[n], n, t, e) : void 0;
    c === void 0 && (c = e[n]), i ? Yn(t, n, c) : Wn(t, n, c);
  }
  return t;
}
var Qn = Zn;
function ei(e, r) {
  for (var t = -1, a = Array(e); ++t < e; )
    a[t] = r(t);
  return a;
}
var ri = ei, ti = 9007199254740991, ai = /^(?:0|[1-9]\d*)$/;
function ni(e, r) {
  var t = typeof e;
  return r = r ?? ti, !!r && (t == "number" || t != "symbol" && ai.test(e)) && e > -1 && e % 1 == 0 && e < r;
}
var le = ni, ii = ri, si = se, oi = Z, ci = ce, ui = le, fi = fe, vi = Object.prototype, li = vi.hasOwnProperty;
function pi(e, r) {
  var t = oi(e), a = !t && si(e), i = !t && !a && ci(e), s = !t && !a && !i && fi(e), o = t || a || i || s, n = o ? ii(e.length, String) : [], c = n.length;
  for (var f in e)
    (r || li.call(e, f)) && !(o && // Safari 9 has enumerable `arguments.length` in strict mode.
    (f == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    i && (f == "offset" || f == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    s && (f == "buffer" || f == "byteLength" || f == "byteOffset") || // Skip index properties.
    ui(f, c))) && n.push(f);
  return n;
}
var hi = pi;
function _i(e) {
  var r = [];
  if (e != null)
    for (var t in Object(e))
      r.push(t);
  return r;
}
var gi = _i, $i = p, di = ne, yi = gi, bi = Object.prototype, Oi = bi.hasOwnProperty;
function Ci(e) {
  if (!$i(e))
    return yi(e);
  var r = di(e), t = [];
  for (var a in e)
    a == "constructor" && (r || !Oi.call(e, a)) || t.push(a);
  return t;
}
var Ti = Ci, Ai = hi, ji = Ti, mi = L;
function Pi(e) {
  return mi(e) ? Ai(e, !0) : ji(e);
}
var pe = Pi, xi = Qn, Ii = pe;
function Si(e) {
  return xi(e, Ii(e));
}
var wi = Si, q = te, Mi = sa, Di = ha, Ei = ga, Li = Sa, k = se, J = Z, Fi = Ja, Gi = ce, Hi = M, Bi = p, Ui = on, Ni = fe, X = ve, zi = wi;
function Ri(e, r, t, a, i, s, o) {
  var n = X(e, t), c = X(r, t), f = o.get(c);
  if (f) {
    q(e, t, f);
    return;
  }
  var v = s ? s(n, c, t + "", e, r, o) : void 0, l = v === void 0;
  if (l) {
    var h = J(c), S = !h && Gi(c), F = !h && !S && Ni(c);
    v = c, h || S || F ? J(n) ? v = n : Fi(n) ? v = Ei(n) : S ? (l = !1, v = Mi(c, !0)) : F ? (l = !1, v = Di(c, !0)) : v = [] : Ui(c) || k(c) ? (v = n, k(n) ? v = zi(n) : (!Bi(n) || Hi(n)) && (v = Li(c))) : l = !1;
  }
  l && (o.set(c, v), i(v, c, a, s, o), o.delete(c)), q(e, t, v);
}
var Vi = Ri, Ki = Jt, qi = te, ki = ia, Ji = Vi, Xi = p, Wi = pe, Yi = ve;
function he(e, r, t, a, i) {
  e !== r && ki(r, function(s, o) {
    if (i || (i = new Ki()), Xi(s))
      Ji(e, r, o, t, he, a, i);
    else {
      var n = a ? a(Yi(e, o), s, o + "", e, r, i) : void 0;
      n === void 0 && (n = s), qi(e, o, n);
    }
  }, Wi);
}
var Zi = he;
function Qi(e) {
  return e;
}
var _e = Qi;
function es(e, r, t) {
  switch (t.length) {
    case 0:
      return e.call(r);
    case 1:
      return e.call(r, t[0]);
    case 2:
      return e.call(r, t[0], t[1]);
    case 3:
      return e.call(r, t[0], t[1], t[2]);
  }
  return e.apply(r, t);
}
var rs = es, ts = rs, W = Math.max;
function as(e, r, t) {
  return r = W(r === void 0 ? e.length - 1 : r, 0), function() {
    for (var a = arguments, i = -1, s = W(a.length - r, 0), o = Array(s); ++i < s; )
      o[i] = a[r + i];
    i = -1;
    for (var n = Array(r + 1); ++i < r; )
      n[i] = a[i];
    return n[r] = t(o), ts(e, this, n);
  };
}
var ns = as;
function is(e) {
  return function() {
    return e;
  };
}
var ss = is, os = ss, Y = re, cs = _e, us = Y ? function(e, r) {
  return Y(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: os(r),
    writable: !0
  });
} : cs, fs = us, vs = 800, ls = 16, ps = Date.now;
function hs(e) {
  var r = 0, t = 0;
  return function() {
    var a = ps(), i = ls - (a - t);
    if (t = a, i > 0) {
      if (++r >= vs)
        return arguments[0];
    } else
      r = 0;
    return e.apply(void 0, arguments);
  };
}
var _s = hs, gs = fs, $s = _s, ds = $s(gs), ys = ds, bs = _e, Os = ns, Cs = ys;
function Ts(e, r) {
  return Cs(Os(e, r, bs), e + "");
}
var As = Ts, js = j, ms = L, Ps = le, xs = p;
function Is(e, r, t) {
  if (!xs(t))
    return !1;
  var a = typeof r;
  return (a == "number" ? ms(t) && Ps(r, t.length) : a == "string" && r in t) ? js(t[r], e) : !1;
}
var Ss = Is, ws = As, Ms = Ss;
function Ds(e) {
  return ws(function(r, t) {
    var a = -1, i = t.length, s = i > 1 ? t[i - 1] : void 0, o = i > 2 ? t[2] : void 0;
    for (s = e.length > 3 && typeof s == "function" ? (i--, s) : void 0, o && Ms(t[0], t[1], o) && (s = i < 3 ? void 0 : s, i = 1), r = Object(r); ++a < i; ) {
      var n = t[a];
      n && e(r, n, a, s);
    }
    return r;
  });
}
var Es = Ds, Ls = Zi, Fs = Es, Gs = Fs(function(e, r, t) {
  Ls(e, r, t);
}), Hs = Gs;
const Bs = /* @__PURE__ */ $e(Hs), Us = function(e) {
  const {
    root: r = ":root",
    prefix: t = "",
    token: a = {},
    useDefaultToken: i = !0
  } = e;
  return ye(() => {
    var c;
    const s = i ? Bs(Q, a) : a, o = de(t, s), n = document.createElement("style");
    return n.innerHTML = `
      ${r}{
        ${Object.entries(o).map(([f, v]) => `--${f}:${v};`).join(" ")}
      }
    `, (c = document.querySelector("head")) == null || c.appendChild(n), () => {
      n.remove();
    };
  }, [a, t]), e.children;
};
Us.defaultToken = Q;
export {
  Us as default,
  de as getKumaConfig
};
