// @jd/ant-theme@0.0.1-alpha.1/dist/Font/demo/family.js
import { ErrorWrapper } from "../../components";
import React from 'react';
import styles from "../../Button/demo/index.module.less";
export var Card = function Card(props) {
  var title = props.title,
    content = props.content;
  return /*#__PURE__*/React.createElement("div", {
    className: "rounded-card py-25 bg-bg flex flex-col items-center justify-between text-title transition-all hover:scale-105 hover:shadow-default",
    style: {
      width: '200px',
      height: '120px'
    }
  }, /*#__PURE__*/React.createElement("span", {
    className: "text-12"
  }, title), /*#__PURE__*/React.createElement("span", {
    className: "text-25"
  }, content));
};
/**
 * 字体
 */
export default function () {
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: " whitespace-nowrap"
  }, /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " ").concat(styles.middle, " ")
  }, /*#__PURE__*/React.createElement("thead", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null), /*#__PURE__*/React.createElement("td", null, "MAC-OS"), /*#__PURE__*/React.createElement("td", null, "Windows"), /*#__PURE__*/React.createElement("td", null, "\u4EAC\u4E1C\u54C1\u724C\u5B57\u4F53"), /*#__PURE__*/React.createElement("td", null, "\u7279\u6B8A\u5C55\u793A\u6570\u5B57\u5B57\u4F53"))), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "\u4E2D\u6587\u5B57\u4F53"), /*#__PURE__*/React.createElement("td", {
    style: {
      fontFamily: 'PingFang SC'
    }
  }, /*#__PURE__*/React.createElement(Card, {
    title: "PingFang SC",
    content: "\u4EAC\u4E1C\u96C6\u56E2\u95E8\u6237"
  })), /*#__PURE__*/React.createElement("td", {
    style: {
      fontFamily: 'Microsoft Yahei'
    }
  }, /*#__PURE__*/React.createElement(Card, {
    title: "Microsoft Yahei",
    content: "\u4EAC\u4E1C\u96C6\u56E2\u95E8\u6237"
  })), /*#__PURE__*/React.createElement("td", {
    className: "font-JD"
  }, /*#__PURE__*/React.createElement(Card, {
    title: "\u4EAC\u4E1C\u6717\u6B63\u4F53",
    content: "\u4EAC\u4E1C\u96C6\u56E2\u95E8\u6237"
  })), /*#__PURE__*/React.createElement("td", null)), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "\u4E2D\u6587\u5B57\u4F53"), /*#__PURE__*/React.createElement("td", {
    style: {
      fontFamily: 'Helvetica Neue'
    }
  }, /*#__PURE__*/React.createElement(Card, {
    title: "Helvetica Neue",
    content: "JD.com, Inc."
  })), /*#__PURE__*/React.createElement("td", {
    style: {
      fontFamily: 'Arial'
    }
  }, /*#__PURE__*/React.createElement(Card, {
    title: "Arial",
    content: "JD.com, Inc."
  })), /*#__PURE__*/React.createElement("td", {
    className: "font-JD"
  }, /*#__PURE__*/React.createElement(Card, {
    title: "\u4EAC\u4E1C\u6717\u6B63\u4F53",
    content: "JD.com, Inc."
  })), /*#__PURE__*/React.createElement("td", {
    className: "font-JDZH"
  }, /*#__PURE__*/React.createElement(Card, {
    title: "\u4EAC\u4E1C\u6B63\u9ED1\u4F53",
    content: "\xA5618.00"
  })))))));
}
