// @jd/ant-theme@0.0.1-alpha.1/dist/InputNumber/demo/basic.js
import styles from "../../Button/demo/index.module.less";
import { ErrorWrapper } from "../../components";
import { Input } from "../..";
import React from 'react';

/**
 *
 */
export default function () {
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("table", {
    className: "text-sub  ".concat(styles.table)
  }, /*#__PURE__*/React.createElement("thead", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("th", null), /*#__PURE__*/React.createElement("th", null, "S=28"), /*#__PURE__*/React.createElement("th", null, "M=32"), /*#__PURE__*/React.createElement("th", null, "L=40"), /*#__PURE__*/React.createElement("th", null, "XL=44"))), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "default"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    size: "small",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    size: "large",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    size: "extraLarge",
    placeholder: "\u8BF7\u8F93\u5165"
  }))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "default"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    defaultValue: "\u6587\u672C\u5185\u5BB9",
    size: "small",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    defaultValue: "\u6587\u672C\u5185\u5BB9",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    defaultValue: "\u6587\u672C\u5185\u5BB9",
    size: "large",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    allowClear: true,
    defaultValue: "\u6587\u672C\u5185\u5BB9",
    size: "extraLarge",
    placeholder: "\u8BF7\u8F93\u5165"
  }))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, "Disable"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    disabled: true,
    size: "small",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    disabled: true,
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    disabled: true,
    size: "large",
    placeholder: "\u8BF7\u8F93\u5165"
  })), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Input, {
    disabled: true,
    size: "extraLarge",
    placeholder: "\u8BF7\u8F93\u5165"
  }))))));
}
