// @jd/ant-theme@0.0.1-alpha.1/dist/Menu/demo/inline-collapsed.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { AppstoreOutlined, ContainerOutlined, DesktopOutlined, MailOutlined, MenuFoldOutlined, MenuUnfoldOutlined, PieChartOutlined } from '@ant-design/icons';
import { Button, Menu } from 'antd';
import React, { useState } from 'react';
function getItem(label, key, icon, children, type) {
  return {
    key: key,
    icon: icon,
    children: children,
    label: label,
    type: type
  };
}
var items = [getItem('Option 1', '1', /*#__PURE__*/React.createElement(PieChartOutlined, null)), getItem('Option 2', '2', /*#__PURE__*/React.createElement(DesktopOutlined, null)), getItem('Option 3', '3', /*#__PURE__*/React.createElement(ContainerOutlined, null)), getItem('Navigation One', 'sub1', /*#__PURE__*/React.createElement(MailOutlined, null), [getItem('Option 5', '5'), getItem('Option 6', '6'), getItem('Option 7', '7'), getItem('Option 8', '8')]), getItem('Navigation Two', 'sub2', /*#__PURE__*/React.createElement(AppstoreOutlined, null), [getItem('Option 9', '9'), getItem('Option 10', '10'), getItem('Submenu', 'sub3', null, [getItem('Option 11', '11'), getItem('Option 12', '12')])])];
var App = function App() {
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    collapsed = _useState2[0],
    setCollapsed = _useState2[1];
  var toggleCollapsed = function toggleCollapsed() {
    setCollapsed(!collapsed);
  };
  return /*#__PURE__*/React.createElement("div", {
    style: {
      width: 256
    }
  }, /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    onClick: toggleCollapsed,
    style: {
      marginBottom: 16
    }
  }, collapsed ? /*#__PURE__*/React.createElement(MenuUnfoldOutlined, null) : /*#__PURE__*/React.createElement(MenuFoldOutlined, null)), /*#__PURE__*/React.createElement(Menu, {
    defaultSelectedKeys: ['1'],
    defaultOpenKeys: ['sub1'],
    mode: "inline",
    theme: "dark",
    inlineCollapsed: collapsed,
    items: items
  }));
};
export default App;
