// @jd/ant-theme@0.0.1-alpha.1/dist/Menu/demo/page.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { ErrorWrapper } from "../../components";
import { Resizable } from 're-resizable';
import React, { useState } from 'react';
import MenuDemo from "./menu-demo";
import styles from "./style.module.less";
/**
 * 可拖拽变更宽度
 */
export default function () {
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    inlineCollapsed = _useState2[0],
    setInlineCollapsed = _useState2[1];
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "flex flex-col bg-bg-page rounded-10"
  }, /*#__PURE__*/React.createElement("div", {
    className: "flex items-center px-20 py-15"
  }, /*#__PURE__*/React.createElement("div", {
    className: "text-primary font-JD flex items-center text-30 gap-20"
  }, /*#__PURE__*/React.createElement("span", {
    className: "font-bold"
  }, "My HR"), /*#__PURE__*/React.createElement("span", {
    className: "text-sub border-l-1 border-sub h-20 opacity-60"
  }), /*#__PURE__*/React.createElement("span", {
    className: "text-title"
  }, "\u77E5\u64AD\u7BA1\u7406\u540E\u53F0")), /*#__PURE__*/React.createElement("div", {
    className: "flex-1 "
  }), /*#__PURE__*/React.createElement("img", {
    src: "http://img13.360buyimg.com/ee/jfs/t1/117513/9/27536/58962/61f10e6dEc2f01a22/910293c0c8b5427f.jpg",
    className: "rounded-full w-40 h-40",
    alt: ""
  })), /*#__PURE__*/React.createElement("div", {
    className: "flex-1 flex"
  }, /*#__PURE__*/React.createElement(Resizable, {
    onResize: function onResize(e, direction, ref, d) {
      console.log(d.width, ref.offsetWidth);
      if (ref.offsetWidth > 100) {
        setInlineCollapsed(false);
      } else {
        setInlineCollapsed(true);
        ref.style.width = '76px';
      }
    },
    defaultSize: {
      width: 200,
      height: 'auto'
    },
    minWidth: 76,
    maxWidth: 500,
    enable: {
      top: false,
      right: true,
      bottom: false,
      left: false,
      topRight: false,
      bottomRight: false,
      bottomLeft: false,
      topLeft: false
    },
    handleClasses: {
      right: styles['resizable-select']
    }
  }, /*#__PURE__*/React.createElement(MenuDemo, {
    inlineCollapsed: inlineCollapsed,
    defaultSelectedKeys: ['home']
  })), /*#__PURE__*/React.createElement("div", {
    className: "flex-1 p-15"
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://img11.360buyimg.com/imagetools/jfs/t1/91765/20/46572/182678/65ee753aF9ad923a0/8ab9d0dede674f62.png",
    className: "w-full",
    alt: ""
  })))));
}
