// @jd/ant-theme@0.0.1-alpha.1/dist/Menu/demo/resizable.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { ErrorWrapper } from "../../components";
import { Resizable } from 're-resizable';
import React, { useState } from 'react';
import MenuDemo from "./menu-demo";
import styles from "./style.module.less";
/**
 * 可拖拽变更宽度
 */
export default function () {
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    inlineCollapsed = _useState2[0],
    setInlineCollapsed = _useState2[1];
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "flex gap-30"
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "mb-10 text-12 text-sub"
  }, "\u62D6\u52A8\u53F3\u4FA7\u53D8\u66F4\u5BBD\u5EA6"), /*#__PURE__*/React.createElement(Resizable, {
    defaultSize: {
      width: 200,
      height: 'auto'
    },
    minWidth: 200,
    maxWidth: 500,
    enable: {
      top: false,
      right: true,
      bottom: false,
      left: false,
      topRight: false,
      bottomRight: false,
      bottomLeft: false,
      topLeft: false
    },
    handleClasses: {
      right: styles['resizable-select']
    }
  }, /*#__PURE__*/React.createElement(MenuDemo, {
    defaultSelectedKeys: ['ghglzcd-1'],
    defaultOpenKeys: ['khjl', 'ghgl', 'ghglzcd1']
  }))), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "mb-10 text-12 text-sub"
  }, "\u5C0F\u4E8E\u4E00\u5B9A\u5BBD\u5EA6\u6536\u8D77"), /*#__PURE__*/React.createElement(Resizable, {
    onResize: function onResize(e, direction, ref, d) {
      console.log(d.width, ref.offsetWidth);
      if (ref.offsetWidth > 100) {
        setInlineCollapsed(false);
      } else {
        setInlineCollapsed(true);
        ref.style.width = '76px';
      }
    },
    defaultSize: {
      width: 200,
      height: 'auto'
    },
    minWidth: 76,
    maxWidth: 500,
    enable: {
      top: false,
      right: true,
      bottom: false,
      left: false,
      topRight: false,
      bottomRight: false,
      bottomLeft: false,
      topLeft: false
    },
    handleClasses: {
      right: styles['resizable-select']
    }
  }, /*#__PURE__*/React.createElement(MenuDemo, {
    inlineCollapsed: inlineCollapsed,
    defaultSelectedKeys: ['home']
  })))));
}
