// @jd/ant-theme@0.0.1-alpha.1/dist/Menu/demo/theme.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { AppstoreOutlined, MailOutlined, SettingOutlined } from '@ant-design/icons';
import { Menu, Switch } from 'antd';
import React, { useState } from 'react';
function getItem(label, key, icon, children, type) {
  return {
    key: key,
    icon: icon,
    children: children,
    label: label,
    type: type
  };
}
var items = [getItem('Navigation One', 'sub1', /*#__PURE__*/React.createElement(MailOutlined, null), [getItem('Option 1', '1'), getItem('Option 2', '2'), getItem('Option 3', '3'), getItem('Option 4', '4')]), getItem('Navigation Two', 'sub2', /*#__PURE__*/React.createElement(AppstoreOutlined, null), [getItem('Option 5', '5'), getItem('Option 6', '6'), getItem('Submenu', 'sub3', null, [getItem('Option 7', '7'), getItem('Option 8', '8')])]), getItem('Navigation Three', 'sub4', /*#__PURE__*/React.createElement(SettingOutlined, null), [getItem('Option 9', '9'), getItem('Option 10', '10'), getItem('Option 11', '11'), getItem('Option 12', '12')])];
var App = function App() {
  var _useState = useState('dark'),
    _useState2 = _slicedToArray(_useState, 2),
    theme = _useState2[0],
    setTheme = _useState2[1];
  var _useState3 = useState('1'),
    _useState4 = _slicedToArray(_useState3, 2),
    current = _useState4[0],
    setCurrent = _useState4[1];
  var changeTheme = function changeTheme(value) {
    setTheme(value ? 'dark' : 'light');
  };
  var onClick = function onClick(e) {
    console.log('click ', e);
    setCurrent(e.key);
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Switch, {
    checked: theme === 'dark',
    onChange: changeTheme,
    checkedChildren: "Dark",
    unCheckedChildren: "Light"
  }), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Menu, {
    theme: theme,
    onClick: onClick,
    style: {
      width: 256
    },
    defaultOpenKeys: ['sub1'],
    selectedKeys: [current],
    mode: "inline",
    items: items
  }));
};
export default App;
