// @jd/ant-theme@0.0.1-alpha.1/dist/components/wrapper/wrapper.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { ErrorWrapper } from "./..";
import { Tooltip } from "../..";
import { Debug, DebugActive } from "../icon";
import { isDev } from "../../utils";
import React, { createContext, useState } from 'react';
export var WrapperContext = /*#__PURE__*/createContext({});
/**
 * components demo
 */
var Wrapper = function Wrapper(props) {
  var title = props.title,
    subtitle = props.subtitle,
    antdLink = props.antdLink,
    theme = props.theme,
    group = props.group;
  var _useState = useState(isDev),
    _useState2 = _slicedToArray(_useState, 2),
    debugActive = _useState2[0],
    setDebugActive = _useState2[1];
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement(WrapperContext.Provider, {
    value: {
      debugActive: debugActive
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: "rounded-30 overflow-hidden mb-30"
  }, /*#__PURE__*/React.createElement("div", {
    className: "p-60 pt-30 text-white font-JD relative ".concat(theme ? "bg-title" : 'bg-colorPrimaryActive')
  }, /*#__PURE__*/React.createElement("div", {
    className: "flex"
  }, /*#__PURE__*/React.createElement("div", {
    className: "flex-1 text-20 flex gap-12 font-bold"
  }, /*#__PURE__*/React.createElement("span", null, "CHO EI design PC guidelines"), /*#__PURE__*/React.createElement("span", null, group === null || group === void 0 ? void 0 : group.title)), /*#__PURE__*/React.createElement("span", {
    className: "text-16 opacity-60"
  }, "\u4E3B\u7AD9\u4F53\u9A8C\u8BBE\u8BA1\u90E8-\u7EFC\u5408\u4E1A\u52A1\u8BBE\u8BA1\u7EC4")), /*#__PURE__*/React.createElement("div", {
    className: "flex items-center absolute right-60 pt-10 select-none"
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: "".concat(debugActive ? '隐藏' : '显示', "\u8C03\u8BD5\u4E13\u7528\u6F14\u793A"),
    className: "cursor-pointer text-20 text-white transition opacity-80 hover:opacity-100"
  }, debugActive ? /*#__PURE__*/React.createElement(DebugActive, {
    onClick: function onClick() {
      setDebugActive(!debugActive);
    }
  }) : /*#__PURE__*/React.createElement(Debug, {
    onClick: function onClick() {
      setDebugActive(!debugActive);
    }
  }))), /*#__PURE__*/React.createElement("div", {
    className: "mt-40 text-40 gap-10 flex items-end"
  }, subtitle ? /*#__PURE__*/React.createElement("span", null, subtitle) : null, title ? /*#__PURE__*/React.createElement("span", {
    className: "opacity-60"
  }, title) : null, antdLink ? /*#__PURE__*/React.createElement("a", {
    href: antdLink,
    style: {
      background: 'rgba(255,255,255,0.6)'
    },
    className: "absolute right-60 bottom-30 text-12 flex items-center text-title rounded-full px-10 py-5 gap-5",
    target: "_blank",
    rel: "noreferrer"
  }, /*#__PURE__*/React.createElement("img", {
    style: {
      height: '20px'
    },
    src: "https://storage.360buyimg.com/imgtools/d581da8fba-a50e42b0-b9d2-11ee-81f6-fd5e8f8402f9.svg",
    alt: ""
  }), "antd\u6587\u6863\u94FE\u63A5") : null)), /*#__PURE__*/React.createElement("div", {
    className: "bg-white p-60 pt-0 overflow-hidden"
  }, props.children))));
};
Wrapper.defaultProps = {};
export default Wrapper;
