// @kuma-ui/core@1.5.7/dist/registry/sheet/ClientStyleSheet.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ClientStyleSheet_exports = {};
__export(ClientStyleSheet_exports, {
  ClientStyleSheet: () => ClientStyleSheet
});
module.exports = __toCommonJS(ClientStyleSheet_exports);
var import_isProduction = require("../../utils/isProduction");
class ClientStyleSheet {
  name;
  deletedRulePlaceholder;
  speedy;
  tags;
  injected;
  rulesCount;
  constructor(name, speedy = false) {
    this.name = name;
    this.deletedRulePlaceholder = `#${name}-deleted-rule{}`;
    this.speedy = speedy;
    this.tags = [];
    this.injected = false;
    this.rulesCount = 0;
  }
  inject() {
    if (this.injected) {
      throw new Error("ClientStyleSheet: sheet already injected");
    }
    if (this.speedy) {
      this.tags[0] = this.makeStyleTag();
      this.speedy = "insertRule" in this.getLatestSheet();
      if (!this.speedy) {
        if (!import_isProduction.isProduction) {
          console.warn(
            "ClientStyleSheet: speedy mode not supported falling back to standard mode."
          );
        }
        this.flush();
      }
    }
    this.injected = true;
  }
  isSpeedy() {
    return this.speedy;
  }
  setSpeedy(bool) {
    if (this.rulesCount === 0) {
      throw new Error(
        "ClientStyleSheet: speedy cannot be when rules have already been inserted"
      );
    }
    this.flush();
    this.speedy = bool;
    this.inject();
  }
  insertRule(rule, index) {
    if (this.speedy) {
      const sheet = this.getLatestSheet();
      if (typeof index !== "number") {
        index = sheet.cssRules.length;
      }
      try {
        sheet.insertRule(rule, index);
      } catch (error) {
        if (!import_isProduction.isProduction) {
          console.warn(
            `ClientStyleSheet: illegal rule: 

${rule}

See https://stackoverflow.com/q/20007992 for more info`
          );
        }
        return -1;
      }
      return this.rulesCount++;
    }
    const insertionPoint = index ? this.tags[index] : void 0;
    this.tags = this.tags.concat(this.makeStyleTag(rule, insertionPoint));
    return this.rulesCount++;
  }
  deleteRule(index) {
    if (index < 0) {
      return;
    }
    if (this.speedy) {
      const sheet = this.getLatestSheet();
      sheet.deleteRule(index);
      sheet.insertRule(this.deletedRulePlaceholder, index);
      return;
    }
    const tag = this.tags[index];
    if (!tag) {
      throw new Error(`ClientStyleSheet: rule at index \`${index}\` not found`);
    }
    tag.parentNode?.removeChild(tag);
    delete this.tags[index];
  }
  flush() {
    this.injected = false;
    this.rulesCount = 0;
    this.tags.forEach((tag) => tag && tag.parentNode?.removeChild(tag));
    this.tags = [];
  }
  cssRules() {
    return this.tags.reduce(
      (rules, tag) => {
        if (tag) {
          return rules.concat(
            Array.from(
              this.getSheet(tag).cssRules,
              (rule) => rule.cssText === this.deletedRulePlaceholder ? void 0 : rule
            )
          );
        }
        return rules;
      },
      []
    );
  }
  makeStyleTag(cssString, relativeToTag) {
    const tag = document.createElement("style");
    tag.setAttribute(`data-${this.name}`, "");
    const nonce = document.querySelector('meta[property="csp-nonce"]')?.getAttribute("content");
    if (nonce) {
      tag.setAttribute("nonce", nonce);
    }
    if (cssString) {
      tag.appendChild(document.createTextNode(cssString));
    }
    const head = document.head || document.getElementsByTagName("head")[0];
    if (relativeToTag) {
      head.insertBefore(tag, relativeToTag);
    } else {
      head.appendChild(tag);
    }
    return tag;
  }
  getSheet(tag) {
    if (tag.sheet) {
      return tag.sheet;
    }
    const sheet = Array.from(document.styleSheets).find(
      (sheet2) => sheet2.ownerNode === tag
    );
    if (sheet) {
      return sheet;
    }
    throw new Error("ClientStyleSheet: styleSheet not found");
  }
  getLatestSheet() {
    const latestTag = this.tags[this.tags.length - 1];
    if (!latestTag) {
      throw new Error("ClientStyleSheet: style tag not found");
    }
    return this.getSheet(latestTag);
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ClientStyleSheet
});
