// @jd/ant-theme@0.0.1-alpha.1/dist/Arrow/index.js
import { ErrorWrapper } from "../components";
import { getProps } from "../utils/PropsType";
import React from 'react';
import ArrowIcon from "./arrow-icon";
import "./index.less";
/**
 * 对应关系
 */
var positionEnum = /*#__PURE__*/function (positionEnum) {
  positionEnum["left"] = "right";
  positionEnum["top"] = "bottom";
  positionEnum["right"] = "left";
  positionEnum["bottom"] = "top";
  return positionEnum;
}(positionEnum || {});
/**
 * 类似ant menu arrow动画
 */
var Arrow = function Arrow(props) {
  var _props$position = props.position,
    position = _props$position === void 0 ? 'bottom' : _props$position,
    _props$active = props.active,
    active = _props$active === void 0 ? false : _props$active,
    size = props.size;
  var wrapperProps = getProps(props, {
    className: ["ant-theme-arrow", // @ts-ignore
    "ant-theme-arrow-".concat(active ? positionEnum[position] : position)]
  }, ['position']);
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("span", wrapperProps, /*#__PURE__*/React.createElement(ArrowIcon, null)));
};
Arrow.defaultProps = {};
export default Arrow;
