// @jd/ant-theme@0.0.1-alpha.1/dist/Color/demo/colors.js
import { ErrorWrapper } from "../../components";
import { theme } from "../..";
import kebabCase from 'lodash/kebabCase';
import React from 'react';
var useToken = theme.useToken;
var colors = ['Neutral', 'Gray', 'Green', 'Cyan', 'Sky blue', 'Blue', 'Indigo', 'Purple', 'Fuchsia', 'Pink', 'Red', 'Orange', 'Yellow'];
var Card = function Card(props) {
  var num = props.num,
    color = props.color;
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "rounded-8 hover:shadow-default transition-all cursor-pointer hover:scale-105",
    style: {
      width: '186px',
      height: '100px',
      // @ts-ignore
      background: color
    }
  }), /*#__PURE__*/React.createElement("div", {
    className: "flex justify-between text-12 mt-4"
  }, /*#__PURE__*/React.createElement("span", null, num), /*#__PURE__*/React.createElement("span", null, color)));
};
/**
 * 色板合集
 */
export default function () {
  var _useToken = useToken(),
    token = _useToken.token;
  console.log(token, 8888);
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "mt-30 flex flex-col gap-y-50"
  }, colors.map(function (item, index) {
    var name = "colors-".concat(kebabCase(item));
    var arr = new Array(10).fill(null);
    // @ts-ignore
    var halfColor = token === null || token === void 0 ? void 0 : token["".concat(name, "-50")];
    return /*#__PURE__*/React.createElement("div", {
      key: index,
      className: "flex"
    }, /*#__PURE__*/React.createElement("div", {
      style: {
        width: '110px'
      }
    }, /*#__PURE__*/React.createElement("div", {
      className: "text-14 font-bold"
    }, item), /*#__PURE__*/React.createElement("span", {
      className: "text-12 text-sub"
    }, "10 colors")), /*#__PURE__*/React.createElement("div", {
      className: "flex-1 flex flex-col"
    }, halfColor ? /*#__PURE__*/React.createElement("div", {
      className: "mb-12 flex"
    }, /*#__PURE__*/React.createElement(Card, {
      color: halfColor,
      num: 50
    })) : null, /*#__PURE__*/React.createElement("div", {
      className: "flex flex-wrap gap-12"
    }, arr.map(function (item1, index1) {
      // @ts-ignore
      var color = token === null || token === void 0 ? void 0 : token["".concat(name, "-").concat((index1 + 1) * 100)];
      return /*#__PURE__*/React.createElement(Card, {
        key: index1,
        color: color,
        num: (index1 + 1) * 100
      });
    }))));
  })));
}
