// @jd/ant-theme@0.0.1-alpha.1/dist/Tabs/demo/ant-custom-tab-bar.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import { Tabs, theme } from "../..";
import React from 'react';
import StickyBox from 'react-sticky-box';
var items = new Array(3).fill(null).map(function (_, i) {
  var id = String(i + 1);
  return {
    label: "Tab ".concat(id),
    key: id,
    children: "Content of Tab Pane ".concat(id),
    style: i === 0 ? {
      height: 200
    } : undefined
  };
});
var App = function App() {
  var _theme$useToken = theme.useToken(),
    colorBgContainer = _theme$useToken.token.colorBgContainer;
  var renderTabBar = function renderTabBar(props, DefaultTabBar) {
    return /*#__PURE__*/React.createElement(StickyBox, {
      offsetTop: 64,
      offsetBottom: 20,
      style: {
        zIndex: 1
      }
    }, /*#__PURE__*/React.createElement(DefaultTabBar, _extends({}, props, {
      style: {
        background: colorBgContainer
      }
    })));
  };
  return /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    renderTabBar: renderTabBar,
    items: items
  });
};
export default App;
