// @jd/ant-theme@0.0.1-alpha.1/dist/Tabs/demo/ant-size.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { Radio, Tabs } from "../..";
import React, { useState } from 'react';
// TODO 大小控制
var App = function App() {
  var _useState = useState('small'),
    _useState2 = _slicedToArray(_useState, 2),
    size = _useState2[0],
    setSize = _useState2[1];
  var onChange = function onChange(e) {
    setSize(e.target.value);
  };
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(Radio.Group, {
    value: size,
    onChange: onChange,
    style: {
      marginBottom: 16
    }
  }, /*#__PURE__*/React.createElement(Radio.Button, {
    value: "small"
  }, "Small"), /*#__PURE__*/React.createElement(Radio.Button, {
    value: "middle"
  }, "Middle"), /*#__PURE__*/React.createElement(Radio.Button, {
    value: "large"
  }, "Large")), /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    size: size,
    style: {
      marginBottom: 32
    },
    items: new Array(3).fill(null).map(function (_, i) {
      var id = String(i + 1);
      return {
        label: "Tab ".concat(id),
        key: id,
        children: "Content of tab ".concat(id)
      };
    })
  }), /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    type: "card",
    size: size,
    items: new Array(3).fill(null).map(function (_, i) {
      var id = String(i + 1);
      return {
        label: "Card Tab ".concat(id),
        key: id,
        children: "Content of card tab ".concat(id)
      };
    })
  }));
};
export default App;
