// @jd/ant-theme@0.0.1-alpha.1/dist/Tabs/demo/class.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { ErrorWrapper } from "../../components";
import { Radio, Segmented, Tabs } from "../..";
import React from 'react';
import styles from "../../Button/demo/index.module.less";
var items = [{
  key: '1',
  label: '选项卡'
}, {
  key: '2',
  label: '选项卡'
}, {
  key: '32',
  label: '选项卡 32'
}, {
  key: '3',
  label: '选项卡'
}];

/**
 *  基础按钮
 */
export default function () {
  return /*#__PURE__*/React.createElement(ErrorWrapper, null, /*#__PURE__*/React.createElement("div", {
    className: "flex flex-col gap-30"
  }, /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full text-left")
  }, /*#__PURE__*/React.createElement("colgroup", null, /*#__PURE__*/React.createElement("col", {
    style: {
      width: 50
    }
  }), /*#__PURE__*/React.createElement("col", {
    style: {
      width: 150
    }
  })), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "border-r-1 border-border whitespace-nowrap",
    rowSpan: 2
  }, "\u4E00\u7EA7tab"), /*#__PURE__*/React.createElement("td", {
    className: "whitespace-nowrap"
  }, "\u4E3B\u8981\u7528\u4E8E\u9875\u5361\u5207\u6362"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    items: items
  }), /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    block: true,
    items: items
  }))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "whitespace-nowrap"
  }, "\u65E0\u9700\u5F3A\u8C03\u7684\u9875\u5361\u5207\u6362", /*#__PURE__*/React.createElement("br", null), " \u53EF\u4E0E\u5F3A\u8C03\u6837\u5F0F\u7EC4\u5408\u4F7F\u7528"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    weak: true,
    items: items
  }), /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    weak: true,
    block: true,
    items: items
  }))))), /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full text-left")
  }, /*#__PURE__*/React.createElement("colgroup", null, /*#__PURE__*/React.createElement("col", {
    style: {
      width: 50
    }
  }), /*#__PURE__*/React.createElement("col", {
    style: {
      width: 150
    }
  })), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "border-r-1 border-border whitespace-nowrap",
    rowSpan: 2
  }, "\u4E8C\u7EA7tab"), /*#__PURE__*/React.createElement("td", {
    className: "whitespace-nowrap"
  }, "\u4E3B\u8981\u7528\u4E8E\u7C7B\u578B\u5207\u6362\u3001", /*#__PURE__*/React.createElement("br", null), "\u7B5B\u9009\u3001\u591A\u9879\u7B5B\u9009\u7B49"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    type: "button",
    items: items
  }), /*#__PURE__*/React.createElement(Tabs, {
    defaultActiveKey: "1",
    type: "button",
    block: true,
    items: items
  }))))), /*#__PURE__*/React.createElement("table", {
    className: "text-sub ".concat(styles.table, " w-full text-left")
  }, /*#__PURE__*/React.createElement("colgroup", null, /*#__PURE__*/React.createElement("col", {
    style: {
      width: 50
    }
  }), /*#__PURE__*/React.createElement("col", {
    style: {
      width: 150
    }
  })), /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "border-r-1 border-border whitespace-nowrap",
    rowSpan: 2
  }, "\u4E09\u7EA7tab"), /*#__PURE__*/React.createElement("td", {
    className: "whitespace-nowrap"
  }, "\u4E3B\u8981\u7528\u4E8E\u8282\u70B9\u5207\u6362"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Segmented, {
    className: "mb-20",
    defaultValue: "1"
    // @ts-ignore
    ,
    options: items.map(function (e) {
      return _objectSpread(_objectSpread({}, e), {}, {
        value: e.key
      });
    })
  }), /*#__PURE__*/React.createElement(Segmented, {
    block: true,
    defaultValue: "1"
    // @ts-ignore
    ,
    options: items.map(function (e) {
      return _objectSpread(_objectSpread({}, e), {}, {
        value: e.key
      });
    })
  }))), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
    className: "whitespace-nowrap"
  }, "\u6309\u94AE\u6837\u5F0F\u7684\u5355\u9009\u7EC4\u5408", /*#__PURE__*/React.createElement("br", null), "\u4E3B\u8981\u7528\u4E8E\u4E92\u65A5\u5207\u6362", /*#__PURE__*/React.createElement("br", null), "\u53CA\u5C0F\u6A21\u5757\u7684tab\u5C55\u793A"), /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Radio.Group, {
    defaultValue: '1',
    size: "large"
  }, items.map(function (item) {
    return /*#__PURE__*/React.createElement(Radio.Button, {
      key: item.key,
      value: item.key
    }, item.label);
  }))))))));
}
