// @kuma-ui/core@1.5.7/dist/components/Spacer/handler.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var handler_exports = {};
__export(handler_exports, {
  defaultSpacerTag: () => defaultSpacerTag,
  isSpacerProps: () => isSpacerProps,
  spacerDefaultProps: () => spacerDefaultProps,
  spacerHandler: () => spacerHandler
});
module.exports = __toCommonJS(handler_exports);
var import_system2 = require("@kuma-ui/system");
const defaultSpacerTag = "div";
const spacerSpecificProps = [
  "horizontal",
  "size"
];
const isSpacerProps = (propName) => {
  return spacerSpecificProps.some((k) => k === propName);
};
const spacerDefaultProps = {};
const spacerHandler = (props) => {
  if (!props.horizontal && !props.size)
    return {};
  const px = typeof props.size === "undefined" ? "0px" : (0, import_system2.toCssUnit)("width", props.size);
  return props.horizontal ? {
    width: px,
    height: "auto",
    display: "inline-block",
    flexShrink: 0
  } : { width: "auto", height: px, display: "block", flexShrink: 0 };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  defaultSpacerTag,
  isSpacerProps,
  spacerDefaultProps,
  spacerHandler
});
